// @vkontakte/vkui/dist/components/SimpleGrid/SimpleGrid.js
import { jsx as _jsx } from "react/jsx-runtime";
import { classNames } from "@vkontakte/vkjs";
import { calculateGap, columnGapClassNames, resolveLayoutProps, rowGapClassNames } from "../../lib/layouts/index.js";
import { RootComponent } from "../RootComponent/RootComponent.js";
const marginClassNames = {
    'auto': "vkuiSimpleGrid__marginAuto",
    'auto-inline': "vkuiSimpleGrid__marginAutoInline",
    'auto-block': "vkuiSimpleGrid__marginAutoBlock"
};
const alignClassNames = {
    start: "vkuiSimpleGrid__alignStart",
    end: "vkuiSimpleGrid__alignEnd",
    center: "vkuiSimpleGrid__alignCenter",
    stretch: "vkuiSimpleGrid__alignStretch",
    baseline: "vkuiSimpleGrid__alignBaseline"
};
const displayClassNames = {
    'none': "vkuiSimpleGrid__displayNone",
    'inline-grid': "vkuiSimpleGrid__displayInlineGrid"
};
/**
 * @see https://vkui.io/components/simple-grid
 */ export const SimpleGrid = ({ columns = 1, gap = 0, margin = 'none', minColWidth, align = 'stretch', display = 'grid', ...restProps })=>{
    const resolvedProps = resolveLayoutProps(restProps);
    const style = {};
    const [rowGap, columnGap] = calculateGap(gap);
    if (typeof rowGap === 'number') {
        style['--vkui_internal--row_gap'] = `${rowGap}px`;
    }
    if (typeof columnGap === 'number') {
        style['--vkui_internal--column_gap'] = `${columnGap}px`;
    }
    style['--vkui_internal--grid_columns'] = `${columns}`;
    if (minColWidth) {
        style['--vkui_internal--min_col_width'] = `${minColWidth}px`;
    }
    return /*#__PURE__*/ _jsx(RootComponent, {
        baseClassName: classNames("vkuiSimpleGrid__host", margin !== 'none' && marginClassNames[margin], alignClassNames[align], minColWidth && "vkuiSimpleGrid__withMinWidth", typeof columnGap === 'string' && columnGapClassNames[columnGap], typeof rowGap === 'string' && rowGapClassNames[rowGap], display !== 'grid' && displayClassNames[display]),
        baseStyle: style,
        ...resolvedProps
    });
};
