// @vkontakte/vkui/dist/components/SelectionControl/SelectionControlLabel/SelectionControlLabel.js
'use client';
import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
/* eslint-disable jsdoc/require-jsdoc */ import { classNames, hasReactNode } from "@vkontakte/vkjs";
import { useAdaptivity } from "../../../hooks/useAdaptivity.js";
import { RootComponent } from "../../RootComponent/RootComponent.js";
import { Footnote } from "../../Typography/Footnote/Footnote.js";
import { Text } from "../../Typography/Text/Text.js";
import { useSelectionControlContext } from "../SelectionControlContext.js";
const densityClassNames = {
    none: "vkuiSelectionControlLabel__densityNone",
    compact: "vkuiSelectionControlLabel__densityCompact"
};
export function SelectionControlLabel({ children, titleAfter, description, ...restProps }) {
    const { noPadding } = useSelectionControlContext();
    const { density = 'none' } = useAdaptivity();
    return /*#__PURE__*/ _jsxs(RootComponent, {
        baseClassName: classNames("vkuiSelectionControlLabel__host", density !== 'regular' && densityClassNames[density], !noPadding && "vkuiSelectionControlLabel__withPadding"),
        ...restProps,
        children: [
            /*#__PURE__*/ _jsxs("div", {
                className: "vkuiSelectionControlLabel__titleLayout",
                children: [
                    /*#__PURE__*/ _jsx(Text, {
                        className: "vkuiSelectionControlLabel__title",
                        children: children
                    }),
                    /*#__PURE__*/ _jsx("div", {
                        className: "vkuiSelectionControlLabel__titleAfter",
                        children: titleAfter
                    })
                ]
            }),
            hasReactNode(description) && /*#__PURE__*/ _jsx(Footnote, {
                className: "vkuiSelectionControlLabel__description",
                children: description
            })
        ]
    });
}
