// @vkontakte/vkui/dist/cssm/components/Radio/Radio.js
'use client';
import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
import { classNames } from "@vkontakte/vkjs";
import { useMergeProps } from "../../hooks/useMergeProps.js";
import { warnOnce } from "../../lib/warnOnce.js";
import { withLabelClickWrapper } from "../../lib/withLabelClickWrapper.js";
import { SelectionControl } from "../SelectionControl/SelectionControl.js";
import { SelectionControlLabel } from "../SelectionControl/SelectionControlLabel/SelectionControlLabel.js";
import { RadioInput } from "./RadioInput/RadioInput.js";
import styles from "./Radio.module.css";
const warn = warnOnce('Radio');
/**
 * @see https://vkui.io/components/radio
 */ export const Radio = ({ // RadioProps
children, description, titleAfter, labelProps, getRef, className, // Tappable props
hoverMode, activeMode, hasHover, hasActive, focusVisibleMode, // Input props
checked, defaultChecked, disabled, readOnly, required, autoFocus, id, name, value, form, onChange, onFocus, onBlur, slotProps, ...restProps })=>{
    /* istanbul ignore if: не проверяем в тестах */ if (process.env.NODE_ENV === 'development') {
        if (labelProps) {
            warn('Свойство `labelProps` устаревшее, используйте `slotProps={ root: {...} }`');
        }
        if (getRef) {
            warn('Свойство `getRef` устаревшее, используйте `slotProps={ input: { getRootRef: ... } }`');
        }
    }
    const { onClick: onRootClick, ...rootRest } = useMergeProps({
        className: classNames(styles.host, className),
        ...labelProps,
        ...restProps
    }, slotProps?.root);
    const inputRest = useMergeProps({
        getRootRef: getRef,
        checked,
        defaultChecked,
        disabled,
        readOnly,
        required,
        autoFocus,
        id,
        name,
        value,
        form,
        onChange,
        onFocus,
        onBlur
    }, slotProps?.input);
    return /*#__PURE__*/ _jsxs(SelectionControl, {
        hoverMode: hoverMode,
        activeMode: activeMode,
        hasHover: hasHover,
        hasActive: hasActive,
        focusVisibleMode: focusVisibleMode,
        disabled: inputRest.disabled,
        onClick: withLabelClickWrapper(onRootClick),
        ...rootRest,
        children: [
            /*#__PURE__*/ _jsx(RadioInput, {
                slotProps: {
                    input: inputRest
                }
            }),
            /*#__PURE__*/ _jsx(SelectionControlLabel, {
                titleAfter: titleAfter,
                description: description,
                children: children
            })
        ]
    });
};
Radio.Input = RadioInput;
