// @vkontakte/vkui/dist/hooks/useModalManager/components/ContextHolder.js
'use client';
import { jsx as _jsx } from "react/jsx-runtime";
import * as React from "react";
import { ModalRoot } from "../../../components/ModalRoot/ModalRoot.js";
import { useActiveModalProps } from "../helpers/useActiveModalProps.js";
export function ContextHolder({ store, onOverlayClosed: onOverlayClosedProp, onOverlayShowed: onOverlayShowedProp, disableModalOverlay: disableModalOverlayProp, disableOpenAnimation: disableOpenAnimationProp, disableCloseAnimation: disableCloseAnimationProp, ...restProps }) {
    const state = React.useSyncExternalStore(store.subscribe, store.getState, store.getState);
    const activeModalProps = useActiveModalProps(state);
    const disableModalOverlay = disableModalOverlayProp || activeModalProps.disableModalOverlay;
    const disableOpenAnimation = disableOpenAnimationProp || activeModalProps.disableOpenAnimation;
    const disableCloseAnimation = disableCloseAnimationProp || activeModalProps.disableCloseAnimation;
    const onOverlayClosed = React.useCallback(()=>{
        store.setOverlayShowed(false);
        onOverlayClosedProp === null || onOverlayClosedProp === void 0 ? void 0 : onOverlayClosedProp();
    }, [
        store,
        onOverlayClosedProp
    ]);
    const onOverlayShowed = React.useCallback(()=>{
        store.setOverlayShowed(true);
        onOverlayShowedProp === null || onOverlayShowedProp === void 0 ? void 0 : onOverlayShowedProp();
    }, [
        store,
        onOverlayShowedProp
    ]);
    const shouldRender = state.modals.length > 0 || !disableModalOverlay && state.overlayShowed;
    if (!shouldRender) {
        return null;
    }
    return /*#__PURE__*/ _jsx(ModalRoot, {
        ...restProps,
        activeModal: state.activeModal,
        disableModalOverlay: disableModalOverlay,
        disableOpenAnimation: disableOpenAnimation,
        disableCloseAnimation: disableCloseAnimation,
        onOverlayClosed: onOverlayClosed,
        onOverlayShowed: onOverlayShowed,
        children: state.modals.map((modalData)=>{
            switch(modalData.type){
                case 'page':
                case 'card':
                    const Modal = modalData.component;
                    return /*#__PURE__*/ _jsx(Modal, {
                        id: modalData.id,
                        modalProps: modalData.modalProps,
                        ...modalData.additionalProps,
                        update: modalData.update,
                        close: modalData.close
                    }, modalData.id);
            }
        })
    });
}
