// @vkontakte/vkui/dist/cssm/components/CalendarTime/CalendarTime.js
'use client';
import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
import { useRef } from "react";
import * as React from "react";
import { classNames } from "@vkontakte/vkjs";
import { Keys, pressedKey } from "../../lib/accessibility.js";
import { setHours, setMinutes } from "../../lib/date.js";
import { AdaptivityProvider } from "../AdaptivityProvider/AdaptivityProvider.js";
import { Button } from "../Button/Button.js";
import { CalendarTimePicker } from "./CalendarTimePicker.js";
import styles from "./CalendarTime.module.css";
const hours = [];
for(let i = 0; i < 24; i += 1){
    hours.push({
        value: i,
        label: String(i).padStart(2, '0')
    });
}
const minutes = [];
for(let i = 0; i < 60; i += 1){
    minutes.push({
        value: i,
        label: String(i).padStart(2, '0')
    });
}
export const CalendarTime = ({ value, onChange, onDoneButtonClick, changeHoursLabel, changeMinutesLabel, setHours: setHoursFn = setHours, setMinutes: setMinutesFn = setMinutes, isDayDisabled, doneButtonText = 'Готово', doneButtonDisabled = false, doneButtonShow = true, minutesTestId, hoursTestId, doneButtonTestId, DoneButton })=>{
    const hoursInputRef = useRef(null);
    const minutesInputRef = useRef(null);
    const doneButtonRef = useRef(null);
    const localHours = isDayDisabled ? hours.map((hour)=>{
        return {
            ...hour,
            disabled: isDayDisabled(setHoursFn(value, hour.value), true)
        };
    }) : hours;
    const localMinutes = isDayDisabled ? minutes.map((minute)=>{
        return {
            ...minute,
            disabled: isDayDisabled(setMinutesFn(value, minute.value), true)
        };
    }) : minutes;
    const onPickerKeyDown = (e)=>{
        const key = pressedKey(e);
        /* Мы хотим иметь возможность быстро, по Enter перемещаться между
     * селектами с часами и минутами, также как мы это делаем по нажатию на Tab */ if (key !== Keys.ENTER) {
            return;
        }
        const steps = [
            hoursInputRef,
            minutesInputRef,
            doneButtonRef
        ].filter((ref)=>Boolean(ref.current));
        const currentStepIndex = steps.findIndex((step)=>step.current === e.target);
        const nextStepIndex = currentStepIndex + 1;
        if (nextStepIndex >= steps.length) {
            return;
        }
        const nextStep = steps[nextStepIndex];
        if (nextStep.current) {
            e.preventDefault();
            nextStep.current?.focus();
        }
    };
    const stopPropagationOfEscapeKeyboardEventWhenSelectIsOpen = React.useCallback((event, isOpen)=>{
        if (isOpen && event.key === 'Escape') {
            event.stopPropagation();
        }
    }, []);
    const onSelectInputKeyDown = (e, isOpen)=>{
        onPickerKeyDown(e);
        stopPropagationOfEscapeKeyboardEventWhenSelectIsOpen(e, isOpen);
    };
    const renderDoneButton = ()=>{
        const ButtonComponent = DoneButton ?? Button;
        return /*#__PURE__*/ _jsx(ButtonComponent, {
            mode: "secondary",
            onClick: onDoneButtonClick,
            size: "l",
            getRootRef: doneButtonRef,
            onKeyDown: onPickerKeyDown,
            disabled: doneButtonDisabled,
            "data-testid": doneButtonTestId,
            children: doneButtonText
        });
    };
    return /*#__PURE__*/ _jsxs("div", {
        className: classNames(styles.host, !doneButtonShow && styles.host__withoutDone),
        children: [
            /*#__PURE__*/ _jsx(CalendarTimePicker, {
                value: value,
                getNumericValue: (v)=>v.getHours(),
                onChange: onChange,
                options: localHours,
                setTime: setHoursFn,
                onInputKeyDown: onSelectInputKeyDown,
                inputRef: hoursInputRef,
                inputLabel: changeHoursLabel,
                inputTestId: hoursTestId
            }),
            /*#__PURE__*/ _jsx("div", {
                className: styles.divider,
                children: ":"
            }),
            /*#__PURE__*/ _jsx(CalendarTimePicker, {
                value: value,
                getNumericValue: (v)=>v.getMinutes(),
                onChange: onChange,
                options: localMinutes,
                setTime: setMinutesFn,
                onInputKeyDown: onSelectInputKeyDown,
                inputRef: minutesInputRef,
                inputLabel: changeMinutesLabel,
                inputTestId: minutesTestId
            }),
            doneButtonShow && /*#__PURE__*/ _jsx("div", {
                className: styles.button,
                children: /*#__PURE__*/ _jsx(AdaptivityProvider, {
                    density: "compact",
                    children: renderDoneButton()
                })
            })
        ]
    });
};
