// @vkontakte/vkui/dist/components/Typography/Footnote/Footnote.js
'use client';
import { jsx as _jsx } from "react/jsx-runtime";
import { classNames } from "@vkontakte/vkjs";
import { useAdaptivity } from "../../../hooks/useAdaptivity.js";
import { Typography } from "../Typography.js";
const densityClassNames = {
    none: "vkuiFootnote__densityNone",
    compact: "vkuiFootnote__densityCompact"
};
export function footnoteClassNames(density, caps = false) {
    return classNames("vkuiFootnote__host", density !== 'regular' && densityClassNames[density], caps && "vkuiFootnote__caps");
}
/**
 * Используется для основных подписей.
 *
 * @see https://vkui.io/components/typography#footnote
 */ export const Footnote = ({ className, caps, Component = 'span', normalize = true, inline = false, ...restProps })=>{
    const { density = 'none' } = useAdaptivity();
    return /*#__PURE__*/ _jsx(Typography, {
        Component: Component,
        normalize: normalize,
        inline: inline,
        className: classNames(className, footnoteClassNames(density, caps)),
        ...restProps
    });
};
