// @vkontakte/vkui/dist/cssm/components/FormLayoutGroup/FormLayoutGroup.js
'use client';
import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
import * as React from "react";
import { classNames } from "@vkontakte/vkjs";
import { useAdaptivity } from "../../hooks/useAdaptivity.js";
import { useExternRef } from "../../hooks/useExternRef.js";
import { Removable } from "../Removable/Removable.js";
import { RootComponent } from "../RootComponent/RootComponent.js";
import styles from "./FormLayoutGroup.module.css";
const densityClassNames = {
    none: classNames(styles.densityNone, 'vkuiInternalFormLayoutGroup--density-none'),
    compact: classNames(styles.densityCompact, 'vkuiInternalFormLayoutGroup--density-compact')
};
/**
 * @see https://vkui.io/components/form-layout-group
 */ export const FormLayoutGroup = ({ children, mode = 'vertical', removable, segmented, noPadding, removePlaceholder = 'Удалить', onRemove, getRootRef, disabled, ...restProps })=>{
    const { density = 'none' } = useAdaptivity();
    const isRemovable = removable && mode === 'horizontal';
    const rootEl = useExternRef(getRootRef);
    return /*#__PURE__*/ _jsx(RootComponent, {
        getRootRef: rootEl,
        Component: "fieldset",
        baseClassName: classNames(styles.host, mode === 'horizontal' && !noPadding && styles.withPadding, density !== 'regular' && densityClassNames[density], mode === 'horizontal' && classNames(styles.modeHorizontal, 'vkuiInternalFormLayoutGroup--mode-horizontal'), mode === 'vertical' && 'vkuiInternalFormLayoutGroup--mode-vertical', isRemovable && classNames(styles.withRemovable, 'vkuiInternalFormLayoutGroup--removable'), segmented && classNames(styles.segmented, 'vkuiInternalFormLayoutGroup--segmented')),
        disabled: disabled,
        ...restProps,
        children: isRemovable ? /*#__PURE__*/ _jsx(Removable, {
            className: styles.removable,
            align: "start",
            removePlaceholder: removePlaceholder,
            onRemove: (e)=>{
                if (rootEl?.current) {
                    onRemove?.(e, rootEl.current);
                }
            },
            disabled: disabled,
            indent: removable === 'indent',
            noPadding: noPadding,
            children: children
        }) : /*#__PURE__*/ _jsxs(React.Fragment, {
            children: [
                children,
                /*#__PURE__*/ _jsx("span", {
                    className: styles.offset,
                    "aria-hidden": true
                })
            ]
        })
    });
};
