// @vkontakte/vkui/dist/cssm/components/ActionSheet/ActionSheetDropdownMenu.js
'use client';
import { jsx as _jsx } from "react/jsx-runtime";
import * as React from "react";
import { classNames } from "@vkontakte/vkjs";
import { useAdaptivityWithJSMediaQueries } from "../../hooks/useAdaptivityWithJSMediaQueries.js";
import { useExternRef } from "../../hooks/useExternRef.js";
import { useGlobalEscKeyDown } from "../../hooks/useGlobalEscKeyDown.js";
import { usePlatform } from "../../hooks/usePlatform.js";
import { isRefObject } from "../../lib/isRefObject.js";
import { stopPropagation } from "../../lib/utils.js";
import { warnOnce } from "../../lib/warnOnce.js";
import { FocusTrap } from "../FocusTrap/FocusTrap.js";
import { Popper } from "../Popper/Popper.js";
import { RootComponent } from "../RootComponent/RootComponent.js";
import { ActionSheetContext } from "./ActionSheetContext.js";
import styles from "./ActionSheet.module.css";
const warn = warnOnce('ActionSheet');
function getEl(ref) {
    return ref && 'current' in ref ? ref.current : ref;
}
export const ActionSheetDropdownMenu = ({ children, toggleRef, closing, onClose: onCloseProp, className, style, popupOffsetDistance = 0, placement, onAnimationStart, onAnimationEnd, allowClickPropagation = false, onClick, getRootRef, // FocusTrap props
autoFocus, restoreFocus, disabled, timeout, ...restProps })=>{
    const platform = usePlatform();
    const { density } = useAdaptivityWithJSMediaQueries();
    const focusTrapRootRef = useExternRef(getRootRef);
    const elementRef = React.useRef(null);
    const { onClose: onActionSheetClose } = React.useContext(ActionSheetContext);
    React.useEffect(()=>{
        if (process.env.NODE_ENV === 'development') {
            const toggleEl = getEl(toggleRef);
            if (!toggleEl) {
                warn(`Свойство "toggleRef" не передано`, 'error');
            }
        }
    }, [
        toggleRef
    ]);
    const targetRef = React.useMemo(()=>{
        if (isRefObject(toggleRef)) {
            return toggleRef;
        }
        return {
            current: toggleRef
        };
    }, [
        toggleRef
    ]);
    const onClose = React.useCallback(()=>{
        onCloseProp?.();
        onActionSheetClose?.('escape-key');
    }, [
        onActionSheetClose,
        onCloseProp
    ]);
    const handleClick = allowClickPropagation ? onClick : (event)=>{
        stopPropagation(event);
        onClick?.(event);
    };
    useGlobalEscKeyDown(true, onClose);
    return /*#__PURE__*/ _jsx(Popper, {
        targetRef: targetRef,
        offsetByMainAxis: popupOffsetDistance,
        placement: placement,
        className: classNames(styles.host, platform === 'ios' && styles.ios, styles.menu, closing ? styles.closing : styles.opening, density === 'compact' && styles.densityCompact, className),
        style: style,
        getRootRef: elementRef,
        usePortal: false,
        onAnimationStart: onAnimationStart,
        onAnimationEnd: onAnimationEnd,
        children: /*#__PURE__*/ _jsx(FocusTrap, {
            autoFocus: autoFocus,
            autoFocusDelay: timeout,
            restoreFocus: restoreFocus,
            disabled: disabled,
            rootRef: focusTrapRootRef,
            children: /*#__PURE__*/ _jsx(RootComponent, {
                ...restProps,
                tabIndex: -1,
                onClick: handleClick,
                getRootRef: focusTrapRootRef,
                children: children
            })
        })
    });
};
