// @vkontakte/vkui/dist/components/Alert/AlertBase.js
'use client';
import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
import * as React from "react";
import { Icon20Cancel } from "@vkontakte/icons";
import { classNames, hasReactNode, noop } from "@vkontakte/vkjs";
import { useAdaptivityWithJSMediaQueries } from "../../hooks/useAdaptivityWithJSMediaQueries.js";
import { useGlobalEscKeyDown } from "../../hooks/useGlobalEscKeyDown.js";
import { usePlatform } from "../../hooks/usePlatform.js";
import { useStableCallback } from "../../hooks/useStableCallback.js";
import { useCSSKeyframesAnimationController } from "../../lib/animation/index.js";
import { stopPropagation } from "../../lib/utils.js";
import { FocusTrap } from "../FocusTrap/FocusTrap.js";
import { IconButton } from "../IconButton/IconButton.js";
import { ModalDismissButton } from "../ModalDismissButton/ModalDismissButton.js";
import { RootComponent } from "../RootComponent/RootComponent.js";
import { AlertActions } from "./AlertActions.js";
import { AlertDescription, AlertTitle } from "./AlertTypography.js";
export const AlertBase = ({ actions, actionsLayout = 'horizontal', children, title, description, onClose: onCloseProp, onClosed, dismissLabel = 'Закрыть предупреждение', renderAction, actionsAlign, dismissButtonMode = 'outside', dismissButtonTestId, onClick, allowClickPropagation = false, titleTestId, descriptionTestId, closing, setClosing, // FocusTrap props
autoFocus, restoreFocus, ...restProps })=>{
    const generatedId = React.useId();
    const titleId = `vkui-alert-${generatedId}-title`;
    const descriptionId = `vkui-alert-${generatedId}-description`;
    const onClose = useStableCallback(onCloseProp || noop);
    const platform = usePlatform();
    const { isDesktop } = useAdaptivityWithJSMediaQueries();
    const itemActionCallbackRef = React.useRef(noop);
    const [animationState, animationHandlers] = useCSSKeyframesAnimationController(closing ? 'exit' : 'enter', {
        onExited () {
            itemActionCallbackRef.current();
            itemActionCallbackRef.current = noop;
            onClosed();
        }
    });
    const isDismissButtonVisible = isDesktop && platform !== 'ios';
    const elementRef = React.useRef(null);
    const close = React.useCallback((reason)=>{
        onClose === null || onClose === void 0 ? void 0 : onClose(reason);
        setClosing === null || setClosing === void 0 ? void 0 : setClosing(true);
    }, [
        onClose,
        setClosing
    ]);
    const onItemClick = React.useCallback((item)=>{
        const { action: itemAction, autoCloseDisabled = false } = item;
        if (autoCloseDisabled) {
            itemAction && itemAction({
                close: ()=>close('click-item')
            });
        } else {
            if (itemAction) {
                itemActionCallbackRef.current = itemAction;
            }
            close('click-item');
        }
    }, [
        close
    ]);
    const handleClick = allowClickPropagation ? onClick : (event)=>{
        stopPropagation(event);
        onClick === null || onClick === void 0 ? void 0 : onClick(event);
    };
    const onEscape = React.useCallback(()=>close('escape-key'), [
        close
    ]);
    const onCloseButtonClick = React.useCallback(()=>close('click-close-button'), [
        close
    ]);
    useGlobalEscKeyDown(true, onEscape);
    return /*#__PURE__*/ _jsx(FocusTrap, {
        rootRef: elementRef,
        autoFocus: autoFocus === undefined ? animationState === 'entered' : autoFocus,
        restoreFocus: restoreFocus,
        children: /*#__PURE__*/ _jsxs(RootComponent, {
            ...animationHandlers,
            tabIndex: -1,
            onClick: handleClick,
            getRootRef: elementRef,
            className: classNames("vkuiAlert__host", platform === 'ios' && "vkuiAlert__ios", platform === 'vkcom' && "vkuiAlert__vkcom", closing ? "vkuiAlert__closing" : "vkuiAlert__opening", isDesktop && "vkuiAlert__desktop"),
            role: "alertdialog",
            "aria-modal": true,
            "aria-labelledby": titleId,
            "aria-describedby": descriptionId,
            ...restProps,
            children: [
                /*#__PURE__*/ _jsxs("div", {
                    className: classNames("vkuiAlert__content", dismissButtonMode === 'inside' && "vkuiAlert__contentWithButton"),
                    tabIndex: -1,
                    children: [
                        hasReactNode(title) && /*#__PURE__*/ _jsx(AlertTitle, {
                            "data-testid": titleTestId,
                            id: titleId,
                            children: title
                        }),
                        hasReactNode(description) && /*#__PURE__*/ _jsx(AlertDescription, {
                            "data-testid": descriptionTestId,
                            id: descriptionId,
                            children: description
                        }),
                        children,
                        isDismissButtonVisible && dismissButtonMode === 'inside' && /*#__PURE__*/ _jsx(IconButton, {
                            label: dismissLabel,
                            className: classNames("vkuiAlert__dismiss", 'vkuiInternalAlert__dismiss'),
                            onClick: onCloseButtonClick,
                            hoverMode: "opacity",
                            activeMode: "opacity",
                            "data-testid": dismissButtonTestId,
                            children: /*#__PURE__*/ _jsx(Icon20Cancel, {})
                        })
                    ]
                }),
                isDismissButtonVisible && dismissButtonMode === 'outside' && /*#__PURE__*/ _jsx(ModalDismissButton, {
                    onClick: onCloseButtonClick,
                    "data-testid": dismissButtonTestId,
                    children: dismissLabel
                }),
                /*#__PURE__*/ _jsx(AlertActions, {
                    actions: actions,
                    actionsAlign: actionsAlign,
                    actionsLayout: actionsLayout,
                    renderAction: renderAction,
                    onItemClick: onItemClick
                })
            ]
        })
    });
};
