// @vkontakte/vkui/dist/components/Typography/Typography.js
import { jsx as _jsx } from "react/jsx-runtime";
import { classNames } from "@vkontakte/vkjs";
import { RootComponent } from "../RootComponent/RootComponent.js";
const stylesWeight = {
    '1': "vkuiTypography__weight1",
    '2': "vkuiTypography__weight2",
    '3': "vkuiTypography__weight3"
};
const stylesAlign = {
    start: "vkuiTypography__alignStart",
    center: "vkuiTypography__alignCenter",
    end: "vkuiTypography__alignEnd"
};
export function weightClassNames(weight, useAccentWeight = false) {
    if (!weight) {
        return '';
    }
    return classNames(stylesWeight[weight], useAccentWeight && "vkuiTypography__accent");
}
export const Typography = ({ weight, useAccentWeight, Component = 'span', normalize, inline, align, ...restProps })=>/*#__PURE__*/ _jsx(RootComponent, {
        Component: Component,
        baseClassName: classNames("vkuiTypography__host", normalize && "vkuiTypography__normalize", inline && "vkuiTypography__inline", weightClassNames(weight, useAccentWeight), align && stylesAlign[align]),
        ...restProps
    });
