// @vkontakte/vkui/dist/components/SubnavigationButton/SubnavigationButton.js
'use client';
import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
import { Icon16Dropdown } from "@vkontakte/icons";
import { classNames } from "@vkontakte/vkjs";
import { useAdaptivity } from "../../hooks/useAdaptivity.js";
import { Tappable } from "../Tappable/Tappable.js";
import { Caption } from "../Typography/Caption/Caption.js";
import { Subhead } from "../Typography/Subhead/Subhead.js";
const appearanceStyles = {
    accent: "vkuiSubnavigationButton__appearanceAccent",
    neutral: "vkuiSubnavigationButton__appearanceNeutral"
};
const modeStyles = {
    primary: "vkuiSubnavigationButton__modePrimary",
    outline: "vkuiSubnavigationButton__modeOutline",
    tertiary: "vkuiSubnavigationButton__modeTertiary"
};
const sizeStyles = {
    s: "vkuiSubnavigationButton__sizeS",
    m: "vkuiSubnavigationButton__sizeM",
    l: "vkuiSubnavigationButton__sizeL"
};
const densityClassNames = {
    none: "vkuiSubnavigationButton__densityNone",
    compact: "vkuiSubnavigationButton__densityCompact"
};
const SubnavigationButtonTypography = ({ textLevel, ...restProps })=>{
    if (textLevel === '1') {
        return /*#__PURE__*/ _jsx(Subhead, {
            ...restProps
        });
    }
    return /*#__PURE__*/ _jsx(Caption, {
        level: textLevel === '2' ? '1' : '2',
        ...restProps
    });
};
/**
 * @see https://vkui.io/components/subnavigation-bar#subnavigation-button
 */ export const SubnavigationButton = ({ mode = 'primary', appearance = 'accent', size = 'm', selected, textLevel = '1', before, after, chevron, children, ...restProps })=>{
    const { density = 'none' } = useAdaptivity();
    return /*#__PURE__*/ _jsx(Tappable, {
        hasActive: false,
        focusVisibleMode: "outside",
        ...restProps,
        baseClassName: classNames("vkuiSubnavigationButton__host", sizeStyles[size], modeStyles[mode], appearanceStyles[appearance], selected && "vkuiSubnavigationButton__selected", density !== 'regular' && densityClassNames[density], restProps.disabled && "vkuiSubnavigationButton__disabled"),
        children: /*#__PURE__*/ _jsxs("span", {
            className: "vkuiSubnavigationButton__in",
            children: [
                before && /*#__PURE__*/ _jsx("span", {
                    className: "vkuiSubnavigationButton__before",
                    children: before
                }),
                /*#__PURE__*/ _jsx(SubnavigationButtonTypography, {
                    textLevel: textLevel,
                    className: "vkuiSubnavigationButton__label",
                    Component: "span",
                    children: children
                }),
                after && /*#__PURE__*/ _jsx("span", {
                    className: "vkuiSubnavigationButton__after",
                    children: after
                }),
                chevron && /*#__PURE__*/ _jsx(Icon16Dropdown, {
                    className: "vkuiSubnavigationButton__chevronIcon"
                })
            ]
        })
    });
};
