// @vkontakte/vkui/dist/cssm/hooks/useSnackbarManager/helpers/createSnackbarStore.js
'use client';
import { addSnackbarToState, closeAllSnackbarsInState, closeOverflowedSnackbarsInState, closeSnackbarInState, removeSnackbarFromState, updateSnackbarPropsInState } from "./snackbarStateHelpers.js";
export const createSnackbarStore = ()=>{
    let state = {
        snackbars: [],
        snackbarsToClose: new Set()
    };
    const listeners = new Set();
    const showedSnackbars = new Set();
    const notify = ()=>{
        listeners.forEach((listener)=>listener());
    };
    const setState = (updater)=>{
        state = updater(state);
        notify();
    };
    return {
        getState: ()=>state,
        getSnackbarsByPlacement: (placement, limit)=>{
            const result = [];
            let openedCount = 0;
            for (const snackbar of state.snackbars){
                if (snackbar.snackbarProps.placement !== placement) {
                    continue;
                }
                const isClosing = state.snackbarsToClose.has(snackbar.id);
                if (openedCount >= limit && !isClosing) {
                    break;
                }
                result.push(snackbar);
                if (!isClosing) {
                    openedCount++;
                }
            }
            return result;
        },
        subscribe: (listener)=>{
            listeners.add(listener);
            return ()=>listeners.delete(listener);
        },
        showedSnackbars,
        addSnackbar: (snackbar)=>{
            setState((oldState)=>addSnackbarToState(oldState, snackbar));
        },
        removeSnackbar: (id)=>{
            setState((oldState)=>removeSnackbarFromState(oldState, id));
            showedSnackbars.delete(id);
        },
        updateSnackbar: (id, config)=>{
            setState((oldState)=>updateSnackbarPropsInState(oldState, id, config));
        },
        closeSnackbar: (id)=>{
            setState((oldState)=>closeSnackbarInState(oldState, id));
        },
        closeAll: (showed)=>{
            setState((oldState)=>closeAllSnackbarsInState(oldState, showed));
        },
        closeOverflowedSnackbars: (placementSnackbars)=>{
            setState((oldState)=>closeOverflowedSnackbarsInState(oldState, placementSnackbars));
        }
    };
};
