// @vkontakte/vkui/dist/cssm/hooks/useSnackbarManager/helpers/useSnackbarActionsWithStore.js
import * as React from "react";
import { randomUUID } from "../../../lib/randomUUID.js";
import { SnackbarWrapper } from "../components/SnackbarWrapper.js";
const resolveMobilePlacement = (placement)=>{
    if (placement.startsWith('top')) {
        return 'top';
    }
    return 'bottom-start';
};
const resolveProps = (props)=>{
    if ('component' in props) {
        return props;
    }
    return {
        component: props
    };
};
export const useSnackbarActionsWithStore = ({ store, limit, queueStrategy, isDesktop })=>{
    const limitRef = React.useRef(limit);
    const queueStrategyRef = React.useRef(queueStrategy);
    const isDesktopRef = React.useRef(isDesktop);
    React.useEffect(()=>{
        limitRef.current = limit;
    }, [
        limit
    ]);
    React.useEffect(()=>{
        queueStrategyRef.current = queueStrategy;
    }, [
        queueStrategy
    ]);
    React.useEffect(()=>{
        isDesktopRef.current = isDesktop;
    }, [
        isDesktop
    ]);
    const update = React.useCallback((id, config)=>{
        store.updateSnackbar(id, config);
    }, [
        store
    ]);
    const close = React.useCallback((id)=>{
        if (store.showedSnackbars.has(id)) {
            store.closeSnackbar(id);
        } else {
            store.removeSnackbar(id);
        }
    }, [
        store
    ]);
    const onOpenSnackbarImpl = React.useCallback((item)=>{
        const placement = item.snackbarProps?.placement || 'bottom-start';
        const resolvedPlacement = isDesktopRef.current ? placement : resolveMobilePlacement(placement);
        const placementSnackbars = store.getSnackbarsByPlacement(resolvedPlacement, limitRef.current);
        const withOverflow = queueStrategyRef.current === 'shift' && placementSnackbars.length >= limitRef.current;
        let resolvePromise;
        const promise = new Promise((resolve)=>{
            resolvePromise = resolve;
        });
        const id = item.id;
        if (withOverflow) {
            store.closeOverflowedSnackbars(placementSnackbars);
        }
        store.addSnackbar({
            ...item,
            id,
            snackbarProps: {
                ...item.snackbarProps,
                id,
                placement: resolvedPlacement,
                onClosed: ()=>{
                    resolvePromise();
                    item.snackbarProps?.onClosed?.();
                }
            }
        });
        return {
            id,
            close: ()=>close(id),
            update: (newProps)=>update(id, newProps),
            onClose: (resolve)=>{
                return promise.then(resolve);
            }
        };
    }, [
        close,
        store,
        update
    ]);
    const openCustom = React.useCallback((config)=>{
        const resolvedProps = resolveProps(config);
        const id = resolvedProps.id || randomUUID();
        return onOpenSnackbarImpl({
            id,
            component: resolvedProps.component,
            snackbarProps: resolvedProps.baseProps,
            additionalProps: resolvedProps.additionalProps,
            close: ()=>close(id),
            update: (newProps)=>update(id, newProps)
        });
    }, [
        close,
        onOpenSnackbarImpl,
        update
    ]);
    const open = React.useCallback((config)=>{
        return openCustom({
            id: config.id,
            component: SnackbarWrapper,
            baseProps: config
        });
    }, [
        openCustom
    ]);
    const closeAllSnackbars = React.useCallback(()=>{
        store.closeAll(store.showedSnackbars);
    }, [
        store
    ]);
    return {
        open,
        openCustom,
        update,
        close,
        closeAll: closeAllSnackbars
    };
};
