// @vkontakte/vkui/dist/components/ChipsInput/ChipsInput.js
'use client';
import { jsx as _jsx } from "react/jsx-runtime";
import { useExternRef } from "../../hooks/useExternRef.js";
import { useMergeProps } from "../../hooks/useMergeProps.js";
import { ChipsInputBase } from "../ChipsInputBase/ChipsInputBase.js";
import { useChipsInput } from "./useChipsInput.js";
/**
 * @see https://vkui.io/components/chips-input
 */ export const ChipsInput = ({ // option
value: valueProp, defaultValue, onChange, // input
getRef, inputValue: inputValueProp, defaultInputValue: defaultInputValueProp, onInputChange: onInputChangeProp, getOptionValue, getOptionLabel, getNewOptionData, disabled: disabledProp, readOnly: readOnlyProp, onFocus: onFocusProp, onBlur: onBlurProp, id: idProp, placeholder: placeholderProp, // other
allowClearButton, delimiter, slotProps, ...restProps })=>{
    const { getRootRef: getInputRef, value: resolvedInputValue, defaultValue: resolvedInputDefaultValue, onChange: resolvedOnInputChange, disabled, readOnly, onFocus, onBlur, id, placeholder, ...inputRest } = useMergeProps({
        getRootRef: getRef,
        value: inputValueProp,
        defaultValue: defaultInputValueProp,
        onChange: onInputChangeProp,
        disabled: disabledProp,
        readOnly: readOnlyProp,
        onFocus: onFocusProp,
        onBlur: onBlurProp,
        id: idProp,
        placeholder: placeholderProp
    }, slotProps === null || slotProps === void 0 ? void 0 : slotProps.input);
    const { value, addOptionFromInput, removeOption, clearOptions, // input
    inputRef: inputRefHook, inputValue, onInputChange } = useChipsInput({
        // option
        value: valueProp,
        defaultValue,
        onChange,
        getOptionLabel,
        getOptionValue,
        getNewOptionData,
        // input
        inputValue: resolvedInputValue,
        defaultInputValue: resolvedInputDefaultValue,
        onInputChange: resolvedOnInputChange,
        // other
        disabled,
        delimiter
    });
    const inputRef = useExternRef(getInputRef, inputRefHook);
    return /*#__PURE__*/ _jsx(ChipsInputBase, {
        value: value,
        clearButtonShown: allowClearButton && (!!value.length || !!inputValue.length),
        onAddChipOption: addOptionFromInput,
        onRemoveChipOption: removeOption,
        onClear: clearOptions,
        slotProps: {
            ...slotProps,
            input: {
                getRootRef: inputRef,
                value: inputValue,
                onChange: onInputChange,
                ...inputRest
            }
        },
        disabled: disabled,
        readOnly: readOnly,
        onFocus: onFocusProp,
        onBlur: onBlur,
        id: id,
        placeholder: placeholder,
        ...restProps
    });
};
