// @vkontakte/vkui/dist/cssm/components/Typography/DisplayTitle/DisplayTitle.js
'use client';
import { jsx as _jsx } from "react/jsx-runtime";
import { classNames } from "@vkontakte/vkjs";
import { useAdaptivity } from "../../../hooks/useAdaptivity.js";
import { Typography } from "../Typography.js";
import styles from "./DisplayTitle.module.css";
const stylesLevel = {
    '1': styles.level1,
    '2': styles.level2,
    '3': styles.level3,
    '4': styles.level4
};
const densityClassNames = {
    none: styles.densityNone,
    compact: styles.densityCompact
};
/**
 * Используется для крупных заголовков.
 *
 * @see https://vkui.io/components/typography#display-title
 */ export const DisplayTitle = ({ className, level = '1', Component = 'span', normalize = true, inline = false, ...restProps })=>{
    const { density = 'none' } = useAdaptivity();
    return /*#__PURE__*/ _jsx(Typography, {
        Component: Component,
        normalize: normalize,
        inline: inline,
        className: classNames(className, density !== 'regular' && densityClassNames[density], stylesLevel[level]),
        ...restProps
    });
};
