// @vkontakte/vkui/dist/components/Typography/Caption/Caption.js
'use client';
import { jsx as _jsx } from "react/jsx-runtime";
import { classNames } from "@vkontakte/vkjs";
import { useAdaptivity } from "../../../hooks/useAdaptivity.js";
import { Typography } from "../Typography.js";
const stylesLevel = {
    '1': "vkuiCaption__level1",
    '2': "vkuiCaption__level2",
    '3': "vkuiCaption__level3"
};
const densityClassNames = {
    none: "vkuiCaption__densityNone",
    compact: "vkuiCaption__densityCompact"
};
export function captionClassNames(density, level = '1', caps = false) {
    return classNames(density !== 'regular' && densityClassNames[density], caps && "vkuiCaption__caps", stylesLevel[level]);
}
/**
 * Используется для мелких подписей.
 *
 * @see https://vkui.io/components/typography#caption
 */ export const Caption = ({ className, level = '1', caps, Component = 'span', normalize = true, inline = false, ...restProps })=>{
    const { density = 'none' } = useAdaptivity();
    return /*#__PURE__*/ _jsx(Typography, {
        Component: Component,
        normalize: normalize,
        inline: inline,
        className: classNames(className, captionClassNames(density, level, caps)),
        ...restProps
    });
};
