// @vkontakte/vkui/dist/components/ToolButton/ToolButton.js
'use client';
import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
import { classNames, hasReactNode } from "@vkontakte/vkjs";
import { useAdaptivity } from "../../hooks/useAdaptivity.js";
import { AdaptiveIconRenderer } from "../AdaptiveIconRenderer/AdaptiveIconRenderer.js";
import { Tappable } from "../Tappable/Tappable.js";
const stylesMode = {
    primary: "vkuiToolButton__modePrimary",
    secondary: "vkuiToolButton__modeSecondary",
    tertiary: "vkuiToolButton__modeTertiary",
    outline: "vkuiToolButton__modeOutline"
};
const stylesAppearance = {
    accent: "vkuiToolButton__appearanceAccent",
    neutral: "vkuiToolButton__appearanceNeutral"
};
const stylesDirection = {
    row: "vkuiToolButton__directionRow",
    column: "vkuiToolButton__directionColumn"
};
const densityClassNames = {
    none: "vkuiToolButton__densityNone",
    regular: "vkuiToolButton__densityRegular"
};
/**
 * Кнопки, которые используются для вызова инструмента, вставки аттачей или
 * для форматирования. Их можно использовать как кнопки для разового действия
 * или для включения/выключения режима.
 *
 * @see https://vkui.io/components/tool-button
 */ export const ToolButton = ({ mode = 'primary', appearance = 'accent', direction = 'row', children, IconCompact, IconRegular, rounded, ...restProps })=>{
    const { density = 'none' } = useAdaptivity();
    const hasChildren = hasReactNode(children);
    return /*#__PURE__*/ _jsxs(Tappable, {
        hoverMode: "vkuiToolButton__hover",
        activeMode: "vkuiToolButton__active",
        Component: restProps.href ? 'a' : 'button',
        focusVisibleMode: "outside",
        baseClassName: classNames("vkuiToolButton__host", rounded && getRoundedClassName(direction, hasChildren), hasChildren && direction === 'row' && "vkuiToolButton__withFakeEndIcon", stylesMode[mode], stylesAppearance[appearance], stylesDirection[direction], density !== 'compact' && densityClassNames[density]),
        ...restProps,
        children: [
            /*#__PURE__*/ _jsx(AdaptiveIconRenderer, {
                IconCompact: IconCompact,
                IconRegular: IconRegular
            }),
            hasChildren && /*#__PURE__*/ _jsx("span", {
                className: "vkuiToolButton__text",
                children: children
            })
        ]
    });
};
export function getRoundedClassName(direction, hasChildren) {
    switch(direction){
        case 'row':
            return "vkuiToolButton__rounded";
        case 'column':
            return hasChildren ? undefined : "vkuiToolButton__rounded";
    }
}
