// @vkontakte/vkui/dist/components/WriteBar/WriteBar.js
'use client';
import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
import * as React from "react";
import { classNames, hasReactNode } from "@vkontakte/vkjs";
import { useExternRef } from "../../hooks/useExternRef.js";
import { useMergeProps } from "../../hooks/useMergeProps.js";
import { usePlatform } from "../../hooks/usePlatform.js";
import { callMultiple } from "../../lib/callMultiple.js";
import { warnOnce } from "../../lib/warnOnce.js";
import { RootComponent } from "../RootComponent/RootComponent.js";
import { useResizeTextarea } from "../Textarea/useResizeTextarea.js";
import { Headline } from "../Typography/Headline/Headline.js";
import { Title } from "../Typography/Title/Title.js";
const warn = warnOnce('WriteBar');
const WriteBarTypography = (props)=>{
    const platform = usePlatform();
    if (platform === 'ios') {
        return /*#__PURE__*/ _jsx(Title, {
            ...props,
            level: "3",
            weight: "3"
        });
    }
    return /*#__PURE__*/ _jsx(Headline, {
        weight: "3",
        ...props
    });
};
/**
 * @see https://vkui.io/components/write-bar
 */ export const WriteBar = ({ // WriteBarProps
before, inlineAfter, after, onHeightChange, shadow = false, getRef, // textarea props
autoComplete, autoCapitalize, autoCorrect, cols, dirName, disabled, maxLength, minLength, name, placeholder, readOnly, required, value: valueProp, wrap, rows, form, onChange: onChangeProp, onFocus, onBlur, id, inputMode, defaultValue, autoFocus, tabIndex, spellCheck, slotProps, ...restProps })=>{
    /* istanbul ignore if: не проверяем в тестах */ if (process.env.NODE_ENV === 'development' && getRef) {
        warn('Свойство `getRef` устаревшее, используйте `slotProps={ textArea: { getRootRef: ... } }`');
    }
    const platform = usePlatform();
    const rootProps = useMergeProps(restProps, slotProps === null || slotProps === void 0 ? void 0 : slotProps.root);
    const { onChange, getRootRef: getTextAreaRef, ...textAreaRest } = useMergeProps({
        className: "vkuiWriteBar__textarea",
        getRootRef: getRef,
        autoComplete,
        autoCapitalize,
        autoCorrect,
        cols,
        dirName,
        disabled,
        maxLength,
        minLength,
        name,
        placeholder,
        readOnly,
        required,
        value: valueProp,
        wrap,
        rows,
        form,
        onChange: onChangeProp,
        onFocus,
        onBlur,
        id,
        inputMode,
        defaultValue,
        autoFocus,
        tabIndex,
        spellCheck
    }, slotProps === null || slotProps === void 0 ? void 0 : slotProps.textArea);
    const [refResizeTextarea, resize] = useResizeTextarea(onHeightChange, true);
    const textareaRef = useExternRef(getTextAreaRef, refResizeTextarea);
    React.useEffect(resize, [
        resize,
        platform
    ]);
    return /*#__PURE__*/ _jsx(RootComponent, {
        baseClassName: classNames("vkuiWriteBar__host", platform === 'ios' && "vkuiWriteBar__ios", shadow && "vkuiWriteBar__shadow"),
        ...rootProps,
        children: /*#__PURE__*/ _jsxs("div", {
            className: "vkuiWriteBar__form",
            children: [
                hasReactNode(before) && /*#__PURE__*/ _jsx("div", {
                    className: "vkuiWriteBar__before",
                    children: before
                }),
                /*#__PURE__*/ _jsxs("div", {
                    className: "vkuiWriteBar__formIn",
                    children: [
                        /*#__PURE__*/ _jsx(WriteBarTypography, {
                            Component: "textarea",
                            onChange: callMultiple(onChange, resize),
                            getRootRef: textareaRef,
                            ...textAreaRest
                        }),
                        hasReactNode(inlineAfter) && /*#__PURE__*/ _jsx("div", {
                            className: "vkuiWriteBar__inlineAfter",
                            children: inlineAfter
                        })
                    ]
                }),
                hasReactNode(after) && /*#__PURE__*/ _jsx("div", {
                    className: "vkuiWriteBar__after",
                    children: after
                })
            ]
        })
    });
};
