// @vkontakte/vkui/dist/components/Flex/FlexItem/FlexItem.js
import { jsx as _jsx } from "react/jsx-runtime";
import { classNames } from "@vkontakte/vkjs";
import { warnOnce } from "../../../lib/warnOnce.js";
import { RootComponent } from "../../RootComponent/RootComponent.js";
const flexClassNames = {
    grow: "vkuiFlexItem__flexGrow",
    shrink: "vkuiFlexItem__flexShrink",
    content: "vkuiFlexItem__flexContent",
    fixed: "vkuiFlexItem__flexFixed"
};
const alignSelfClassNames = {
    start: "vkuiFlexItem__alignSelfStart",
    end: "vkuiFlexItem__alignSelfEnd",
    center: "vkuiFlexItem__alignSelfCenter",
    baseline: "vkuiFlexItem__alignSelfBaseline",
    stretch: "vkuiFlexItem__alignSelfStretch"
};
const warn = warnOnce('Flex.Item');
export const FlexItem = ({ alignSelf, flex, flexBasis, ...restProps })=>{
    if (process.env.NODE_ENV === 'development') {
        warn('Компонент Flex.Item устарел, используйте компонент Flex в качестве альтернативы.');
    }
    return /*#__PURE__*/ _jsx(RootComponent, {
        baseStyle: {
            flexBasis
        },
        baseClassName: classNames(alignSelf && alignSelfClassNames[alignSelf], flex && flexClassNames[flex]),
        ...restProps
    });
};
