// @vkontakte/vkui/dist/cssm/components/Button/Button.js
'use client';
import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
import * as React from "react";
import { classNames, hasReactNode } from "@vkontakte/vkjs";
import { useAdaptivity } from "../../hooks/useAdaptivity.js";
import { usePlatform } from "../../hooks/usePlatform.js";
import { Spinner } from "../Spinner/Spinner.js";
import { Tappable } from "../Tappable/Tappable.js";
import "../Tappable/Tappable.module.css";
import "../Spinner/Spinner.module.css";
import styles from "./Button.module.css";
const stylesElevation = {
    '1': styles.elevation1,
    '2': styles.elevation2,
    '3': styles.elevation3,
    '4': styles.elevation4
};
const stylesSize = {
    s: styles.sizeS,
    m: styles.sizeM,
    l: styles.sizeL
};
const stylesMode = {
    primary: styles.modePrimary,
    secondary: styles.modeSecondary,
    tertiary: styles.modeTertiary,
    outline: styles.modeOutline,
    link: styles.modeLink
};
const stylesAppearance = {
    'accent': styles.appearanceAccent,
    'positive': styles.appearancePositive,
    'negative': styles.appearanceNegative,
    'neutral': styles.appearanceNeutral,
    'overlay': styles.appearanceOverlay,
    'accent-invariable': styles.appearanceAccentInvariable
};
const stylesAlign = {
    left: styles.alignLeft,
    right: styles.alignRight
};
const densityClassNames = {
    none: styles.densityNone,
    regular: styles.densityRegular
};
/**
 * @see https://vkui.io/components/button
 */ export const Button = ({ size = 's', mode = 'primary', appearance = 'accent', stretched = false, align = 'center', children, before, after, getRootRef, loading, loadingLabel = 'Загрузка...', onClick, disableSpinnerAnimation, rounded, disabled, href, 'aria-label': ariaLabelProp, elevation, ...restProps })=>{
    const hasIconOnly = !children && Boolean(after) !== Boolean(before);
    const { density = 'none' } = useAdaptivity();
    const platform = usePlatform();
    const isDisabled = disabled || loading;
    const hasHref = href !== undefined;
    const ariaLabel = loading ? loadingLabel : ariaLabelProp;
    const buttonProps = React.useMemo(()=>{
        if (hasHref) {
            return isDisabled ? {
                // Для disabled/loading ссылок нужно удалить href и добавить role="link"
                // согласно https://w3c.github.io/html-aria/#example-communicate-a-disabled-link-with-aria
                'role': 'link',
                'Component': 'a',
                'aria-disabled': isDisabled
            } : {
                href,
                'aria-disabled': isDisabled
            };
        } else {
            return {
                Component: 'button',
                disabled
            };
        }
    }, [
        disabled,
        hasHref,
        href,
        isDisabled
    ]);
    return /*#__PURE__*/ _jsxs(Tappable, {
        hoverMode: styles.hover,
        activeMode: styles.active,
        focusVisibleMode: "outside",
        "aria-busy": loading,
        hasHover: !loading,
        hasActive: !loading,
        ...buttonProps,
        ...restProps,
        "aria-label": ariaLabel,
        onClick: isDisabled ? undefined : onClick,
        baseClassName: classNames(styles.host, stylesSize[size], stylesMode[mode], stylesAppearance[appearance], align !== 'center' && stylesAlign[align], density !== 'compact' && densityClassNames[density], platform === 'ios' && styles.ios, stretched && styles.stretched, hasIconOnly && !stretched && styles.singleIcon, loading && styles.loading, rounded && styles.rounded, disabled && styles.disabled, elevation && stylesElevation[elevation]),
        getRootRef: getRootRef,
        children: [
            loading && /*#__PURE__*/ _jsx(Spinner, {
                size: "s",
                className: styles.spinner,
                disableAnimation: disableSpinnerAnimation,
                noColor: true,
                "aria-hidden": "true"
            }),
            /*#__PURE__*/ _jsxs("span", {
                className: styles.in,
                children: [
                    hasReactNode(before) && /*#__PURE__*/ _jsx("span", {
                        className: styles.before,
                        role: "presentation",
                        "data-testid": process.env.NODE_ENV === 'test' ? 'before' : undefined,
                        children: before
                    }),
                    hasReactNode(children) && /*#__PURE__*/ _jsx("span", {
                        className: styles.content,
                        "data-testid": process.env.NODE_ENV === 'test' ? 'children' : undefined,
                        children: children
                    }),
                    hasReactNode(after) && /*#__PURE__*/ _jsx("span", {
                        className: styles.after,
                        role: "presentation",
                        "data-testid": process.env.NODE_ENV === 'test' ? 'after' : undefined,
                        children: after
                    })
                ]
            })
        ]
    });
};
