// @vkontakte/vkui/dist/cssm/components/Checkbox/CheckboxSimple/CheckboxSimple.js
'use client';
import { jsx as _jsx } from "react/jsx-runtime";
import { classNames } from "@vkontakte/vkjs";
import { useAdaptivity } from "../../../hooks/useAdaptivity.js";
import { useMergeProps } from "../../../hooks/useMergeProps.js";
import { withLabelClickWrapper } from "../../../lib/withLabelClickWrapper.js";
import { Tappable } from "../../Tappable/Tappable.js";
import { CheckboxInput } from "../CheckboxInput/CheckboxInput.js";
import styles from "./CheckboxSimple.module.css";
const densityClassNames = {
    none: styles.densityNone,
    compact: styles.densityCompact
};
export function CheckboxSimple({ // CheckboxProps
getRef, description, titleAfter, noPadding, children, // CheckboxInputProps
indeterminate, defaultIndeterminate, IconOnCompact, IconOnRegular, IconOffCompact, IconOffRegular, IconIndeterminate, // Tappable props
hoverMode: hoverModeProp, activeMode: activeModeProp, hasHover, hasActive, focusVisibleMode, // Input props
checked, defaultChecked, disabled, readOnly, required, autoFocus, id, name, value, onChange, onFocus, onBlur, slotProps, ...restProps }) {
    const { onClick, ...rootRest } = useMergeProps(restProps, slotProps?.root);
    const inputRest = useMergeProps({
        getRootRef: getRef,
        checked,
        defaultChecked,
        disabled,
        readOnly,
        required,
        autoFocus,
        id,
        name,
        value,
        onChange,
        onFocus,
        onBlur
    }, slotProps?.input);
    const { density = 'none' } = useAdaptivity();
    const hoverMode = hoverModeProp || (noPadding ? 'opacity' : 'background');
    const activeMode = activeModeProp || (noPadding ? 'opacity' : 'background');
    return /*#__PURE__*/ _jsx(Tappable, {
        baseClassName: classNames(styles.host, !noPadding && styles.withPadding, density !== 'regular' && densityClassNames[density]),
        disabled: inputRest.disabled,
        hoverMode: hoverMode,
        activeMode: activeMode,
        hasHover: hasHover,
        hasActive: hasActive,
        focusVisibleMode: focusVisibleMode,
        Component: "label",
        onClick: withLabelClickWrapper(onClick),
        ...rootRest,
        children: /*#__PURE__*/ _jsx(CheckboxInput, {
            indeterminate: indeterminate,
            defaultIndeterminate: defaultIndeterminate,
            IconIndeterminate: IconIndeterminate,
            IconOnCompact: IconOnCompact,
            IconOnRegular: IconOnRegular,
            IconOffCompact: IconOffCompact,
            IconOffRegular: IconOffRegular,
            slotProps: {
                input: inputRest
            }
        })
    });
}
