// @vkontakte/vkui/dist/components/Avatar/AvatarBadge/AvatarBadgeWithPreset.js
'use client';
import { jsx as _jsx } from "react/jsx-runtime";
import * as React from "react";
import { classNames } from "@vkontakte/vkjs";
import { useConfigDirection } from "../../../hooks/useConfigDirection.js";
import { getBadgeIconSizeByImageBaseSize, ImageBase, ImageBaseContext } from "../../ImageBase/ImageBase.js";
import { Icon12Circle, Icon12OnlineMobile } from "./icons.js";
export const AvatarBadgeWithPreset = ({ preset = 'online', className, ...restProps })=>{
    const direction = useConfigDirection();
    const isRtl = direction === 'rtl';
    const { size } = React.useContext(ImageBaseContext);
    const badgeSize = getBadgeIconSizeByImageBaseSize(size);
    const isOnlinePreset = preset === 'online';
    const presetClassName = isOnlinePreset ? "vkuiAvatarBadge__presetOnline" : "vkuiAvatarBadge__presetOnlineMobile";
    const Icon = isOnlinePreset ? Icon12Circle : Icon12OnlineMobile;
    return /*#__PURE__*/ _jsx(ImageBase.Badge, {
        background: "stroke",
        className: classNames("vkuiAvatarBadge__host", isRtl && "vkuiAvatarBadge__rtl", presetClassName, className),
        ...restProps,
        children: /*#__PURE__*/ _jsx(Icon, {
            width: badgeSize,
            height: badgeSize
        })
    });
};
