// @vkontakte/vkui/dist/components/SelectMimicry/SelectMimicry.js
'use client';
import { jsx as _jsx } from "react/jsx-runtime";
import { classNames } from "@vkontakte/vkjs";
import { useAdaptivity } from "../../hooks/useAdaptivity.js";
import { useAutoFocus } from "../../hooks/useAutoFocus.js";
import { useExternRef } from "../../hooks/useExternRef.js";
import { getFormFieldModeFromSelectType } from "../../lib/select.js";
import { DropdownIcon } from "../DropdownIcon/DropdownIcon.js";
import { FormField } from "../FormField/FormField.js";
import { SelectTypography } from "../SelectTypography/SelectTypography.js";
const densityClassNames = {
    none: "vkuiSelect__densityNone",
    compact: "vkuiSelect__densityCompact"
};
/**
 * @see https://vkui.io/components/select-mimicry
 */ export const SelectMimicry = ({ tabIndex = 0, placeholder, children, align, getRootRef, multiline, disabled, onClick, before, after = /*#__PURE__*/ _jsx(DropdownIcon, {}), selectType = 'default', status, className, autoFocus, ...restProps })=>{
    const rootRef = useExternRef(getRootRef);
    const { density = 'none' } = useAdaptivity();
    const title = children || placeholder;
    useAutoFocus(rootRef, autoFocus);
    return /*#__PURE__*/ _jsx(FormField, {
        ...restProps,
        tabIndex: disabled ? undefined : tabIndex,
        className: classNames("vkuiSelect__host", density !== 'regular' && densityClassNames[density], !children && "vkuiSelect__empty", multiline && "vkuiSelect__multiline", align === 'center' && "vkuiSelect__alignCenter", align === 'right' && "vkuiSelect__alignRight", before && "vkuiSelect__hasBefore", className),
        getRootRef: rootRef,
        onClick: disabled ? undefined : onClick,
        disabled: disabled,
        before: before,
        after: after,
        mode: getFormFieldModeFromSelectType(selectType),
        status: status,
        children: /*#__PURE__*/ _jsx("div", {
            className: "vkuiSelect__container",
            children: /*#__PURE__*/ _jsx(SelectTypography, {
                selectType: selectType,
                className: "vkuiSelect__title",
                children: title
            })
        })
    });
};
