// @vkontakte/vkui/dist/hooks/useSnackbarManager/components/SnackbarHolder.js
'use client';
import { jsx as _jsx, Fragment as _Fragment } from "react/jsx-runtime";
import * as React from "react";
import { SnackbarsContainer } from "./SnackbarsContainer.js";
export const SnackbarHolder = ({ store, limit, offsetYStart, offsetYEnd, zIndex })=>{
    const state = React.useSyncExternalStore(store.subscribe, store.getState, store.getState);
    const onSnackbarOpen = React.useCallback((id)=>{
        store.showedSnackbars.add(id);
    }, [
        store
    ]);
    const [snackbarsMap, placements] = React.useMemo(()=>{
        const map = {};
        const placements = [];
        const openedSnackbarsCountersByPlacement = {};
        for (const snackbar of state.snackbars){
            const placement = snackbar.snackbarProps.placement;
            const isClosing = state.snackbarsToClose.has(snackbar.id);
            if (!map[placement]) {
                map[placement] = [];
                openedSnackbarsCountersByPlacement[placement] = 0;
                placements.push(placement);
            }
            const openedCount = openedSnackbarsCountersByPlacement[placement];
            if (openedCount >= limit && !isClosing) {
                continue;
            }
            map[placement].push({
                ...snackbar,
                snackbarProps: {
                    ...snackbar.snackbarProps,
                    open: isClosing ? false : undefined
                }
            });
            if (!isClosing) {
                openedSnackbarsCountersByPlacement[placement] = openedCount + 1;
            }
        }
        return [
            map,
            placements
        ];
    }, [
        state.snackbars,
        state.snackbarsToClose,
        limit
    ]);
    if (placements.length === 0) {
        return null;
    }
    return /*#__PURE__*/ _jsx(_Fragment, {
        children: placements.map((placement)=>/*#__PURE__*/ _jsx(SnackbarsContainer, {
                snackbars: snackbarsMap[placement],
                placement: placement,
                onSnackbarContainerClosed: store.removeSnackbar,
                onSnackbarOpen: onSnackbarOpen,
                offsetYStart: offsetYStart,
                offsetYEnd: offsetYEnd,
                zIndex: zIndex
            }, placement))
    });
};
