// @vkontakte/vkui/dist/hooks/useSnackbarManager/components/SnackbarAnimatedWrapper.js
'use client';
import { jsx as _jsx } from "react/jsx-runtime";
import { useEffect, useRef } from "react";
import { classNames } from "@vkontakte/vkjs";
import { useCSSKeyframesAnimationController } from "../../../lib/animation/index.js";
import { usePlatform } from "../../usePlatform.js";
const animationStateClassNames = {
    enter: "vkuicomponents__stateEnter",
    entering: "vkuicomponents__stateEnter",
    entered: "vkuicomponents__stateEnter",
    exit: "vkuicomponents__stateExit",
    exiting: "vkuicomponents__stateExit",
    exited: undefined
};
const placementClassNames = {
    top: "vkuicomponents__placementTop",
    bottom: "vkuicomponents__placementBottom"
};
export const SnackbarAnimatedWrapper = ({ snackbarItem, close = false, onClosed, placement })=>{
    const ref = useRef(null);
    const platform = usePlatform();
    const [animationState, animationHandlers] = useCSSKeyframesAnimationController(close ? 'exit' : 'enter', {
        onExited: ()=>onClosed(snackbarItem.id)
    });
    useEffect(()=>{
        if (ref.current && animationState === 'entered') {
            ref.current.style.setProperty('--vkui_internal--SnackbarAnimatedWrapper_height', `${ref.current.offsetHeight}px`);
        }
    }, [
        animationState
    ]);
    if (animationState === 'exited') {
        return null;
    }
    return /*#__PURE__*/ _jsx("div", {
        ref: ref,
        className: classNames("vkuicomponents__host", animationStateClassNames[animationState], platform === 'ios' && "vkuicomponents__ios", placementClassNames[placement]),
        ...animationHandlers,
        children: /*#__PURE__*/ _jsx(snackbarItem.component, {
            ...snackbarItem.additionalProps,
            id: snackbarItem.id,
            snackbarProps: snackbarItem.snackbarProps,
            close: snackbarItem.close,
            update: snackbarItem.update
        })
    });
};
