// @vkontakte/vkui/dist/hooks/useSnackbarManager/components/SnackbarsContainer.js
'use client';
import { jsx as _jsx } from "react/jsx-runtime";
import * as React from "react";
import { classNames } from "@vkontakte/vkjs";
import { Flex } from "../../../components/Flex/Flex.js";
import { SnackbarAnimatedWrapper } from "./SnackbarAnimatedWrapper.js";
import { SnackbarsContainerContext } from "./SnackbarsContainerContext.js";
const placementClassNames = {
    'top-start': "vkuicomponents__placementTopStart",
    'top-end': "vkuicomponents__placementTopEnd",
    'top': "vkuicomponents__placementTop",
    'bottom-start': "vkuicomponents__placementBottomStart",
    'bottom-end': "vkuicomponents__placementBottomEnd",
    'bottom': "vkuicomponents__placementBottom"
};
const resolveOffset = (offset)=>{
    if (typeof offset === 'undefined') {
        return '';
    }
    return typeof offset === 'number' ? `${offset}px` : offset;
};
export const SnackbarsContainer = ({ snackbars, placement, onSnackbarOpen, onSnackbarContainerClosed: onSnackbarContainerClosedProp, offsetYStart, offsetYEnd, zIndex })=>{
    const [snackbarsWrappersToClose, setSnackbarsWrappersToClose] = React.useState(new Set());
    const onClosed = React.useCallback((id)=>{
        setSnackbarsWrappersToClose((oldState)=>new Set([
                ...oldState,
                id
            ]));
    }, []);
    const onSnackbarContainerClosed = React.useCallback((id)=>{
        setSnackbarsWrappersToClose((oldState)=>{
            oldState.delete(id);
            return new Set(oldState);
        });
        onSnackbarContainerClosedProp(id);
    }, [
        onSnackbarContainerClosedProp
    ]);
    const contextValue = React.useMemo(()=>({
            isInsideContainer: true,
            onClosed,
            onOpen: onSnackbarOpen
        }), [
        onClosed,
        onSnackbarOpen
    ]);
    const containerStyles = {
        '--vkui_internal--snackbars_container_offset_y_start': resolveOffset(offsetYStart),
        '--vkui_internal--snackbars_container_offset_y_end': resolveOffset(offsetYEnd),
        ...zIndex !== undefined && {
            zIndex
        }
    };
    return /*#__PURE__*/ _jsx(SnackbarsContainerContext.Provider, {
        value: contextValue,
        children: /*#__PURE__*/ _jsx(Flex, {
            className: classNames("vkuicomponents__host", placementClassNames[placement]),
            style: containerStyles,
            children: snackbars.map((snackbar)=>/*#__PURE__*/ _jsx(SnackbarAnimatedWrapper, {
                    snackbarItem: snackbar,
                    onClosed: onSnackbarContainerClosed,
                    close: snackbarsWrappersToClose.has(snackbar.id),
                    placement: placement.startsWith('top') ? 'top' : 'bottom'
                }, snackbar.id))
        })
    });
};
