// @vkontakte/vkui/dist/hooks/useModalManager/helpers/createModalStore.js
'use client';
import { addModalToState, closeAllModals, closeModal, removeModalFromState, setOverlayShowedInState, setPrevActiveModal, updateModalPropsInState } from "./modalStateHelpers.js";
export const createModalStore = ()=>{
    let state = {
        activeModal: null,
        modals: [],
        overlayShowed: false
    };
    const listeners = new Set();
    const needCloseModals = new Set();
    const notify = ()=>{
        listeners.forEach((listener)=>listener());
    };
    const setState = (updater)=>{
        const newState = updater(state);
        if (newState !== state) {
            state = newState;
            notify();
        }
    };
    return {
        getState: ()=>state,
        subscribe: (listener)=>{
            listeners.add(listener);
            return ()=>listeners.delete(listener);
        },
        needCloseModals,
        addModal: (modalData)=>{
            setState((oldState)=>addModalToState(oldState, modalData));
        },
        removeModal: (id)=>{
            setState((oldState)=>removeModalFromState(oldState, id));
        },
        updateModalProps: (id, props)=>{
            setState((oldState)=>updateModalPropsInState(oldState, id, props));
        },
        closeModal: (id)=>{
            setState((oldState)=>closeModal(oldState, id, needCloseModals));
        },
        closeAll: ()=>{
            setState((oldState)=>closeAllModals(oldState, needCloseModals));
        },
        setPrevActive: (id)=>{
            setState((oldState)=>setPrevActiveModal(oldState, id, needCloseModals));
        },
        closePrevActiveIfNoHistory: ()=>{
            setState((oldState)=>{
                if (oldState.activeModal) {
                    return setPrevActiveModal(oldState, oldState.activeModal, needCloseModals);
                }
                return oldState;
            });
        },
        setOverlayShowed: (value)=>{
            setState((oldState)=>setOverlayShowedInState(oldState, value));
        }
    };
};
