// @vkontakte/vkui/dist/hooks/useModalManager/helpers/modalStateHelpers.js
export const setOverlayShowedInState = (state, overlayShowed)=>{
    if (state.overlayShowed === overlayShowed) {
        return state;
    }
    return {
        ...state,
        overlayShowed
    };
};
export const removeModalFromState = (state, id)=>{
    const { modals } = state;
    const newModals = modals.filter(({ id: modalId })=>id !== modalId);
    return {
        ...state,
        modals: newModals
    };
};
export const setPrevActiveModal = (state, id, needCloseModals)=>{
    const { modals, activeModal } = state;
    let newActiveModal = activeModal;
    needCloseModals.add(id);
    const activeModalIndex = modals.findIndex(({ id: modalId })=>id === modalId);
    if (activeModal === id && activeModalIndex === modals.length - 1) {
        if (modals.length === 1) {
            newActiveModal = null;
        } else {
            newActiveModal = modals[activeModalIndex - 1].id || null;
        }
    }
    return {
        ...state,
        activeModal: newActiveModal,
        modals
    };
};
export const updateModalPropsInState = (state, id, props)=>{
    const { modals } = state;
    const cleanProps = {
        ...props
    };
    if ('id' in cleanProps) {
        delete cleanProps['id'];
    }
    const modalIndex = modals.findIndex((modal)=>modal.id === id);
    if (modalIndex === -1) {
        return state;
    }
    const currentModal = modals[modalIndex];
    const newModalProps = (()=>{
        switch(currentModal.type){
            case 'page':
            case 'card':
                return {
                    ...currentModal,
                    modalProps: {
                        ...currentModal.modalProps,
                        ...cleanProps
                    }
                };
        }
    })();
    return {
        ...state,
        modals: [
            ...modals.slice(0, modalIndex),
            newModalProps,
            ...modals.slice(modalIndex + 1)
        ]
    };
};
export const closeAllModals = (state, needCloseModals)=>{
    const { modals, activeModal } = state;
    const newModals = modals.filter(({ id: modalId })=>activeModal === modalId);
    if (activeModal) {
        needCloseModals.add(activeModal);
    }
    return {
        ...state,
        modals: newModals,
        activeModal: null
    };
};
/**
 * Закрывает модал: если активный - устанавливает предыдущий, иначе просто удаляет
 */ export const closeModal = (state, id, needCloseModals)=>{
    const { activeModal } = state;
    if (!activeModal) {
        return state;
    }
    if (activeModal === id) {
        return setPrevActiveModal(state, activeModal, needCloseModals);
    } else {
        return removeModalFromState(state, id);
    }
};
export const addModalToState = (state, modalData)=>{
    if (state.modals.find((modal)=>modal.id === modalData.id)) {
        return state;
    }
    return {
        ...state,
        modals: [
            ...state.modals,
            modalData
        ],
        activeModal: modalData.id || null
    };
};
export const getActiveModalProp = (state, propGetter)=>{
    if (!state.activeModal) {
        return undefined;
    }
    const activeModalProps = state.modals.find((modal)=>modal.id === state.activeModal);
    if (!activeModalProps) {
        return undefined;
    }
    return propGetter(activeModalProps);
};
