// @vkontakte/vkui/dist/hooks/useModalManager/helpers/useModalActions.js
import * as React from "react";
import { randomUUID } from "../../../lib/randomUUID.js";
import { ModalCardWrapper } from "../components/ModalCardWrapper.js";
import { ModalPageWrapper } from "../components/ModalPageWrapper.js";
const createModalCallbacks = (id, modalProps, store, resolvePromise)=>{
    const onClose = (reason, event)=>{
        var _modalProps_onClose;
        store.setPrevActive(id);
        (_modalProps_onClose = modalProps.onClose) === null || _modalProps_onClose === void 0 ? void 0 : _modalProps_onClose.call(modalProps, reason, event);
    };
    const onClosed = ()=>{
        var _modalProps_onClosed;
        if (store.needCloseModals.has(id)) {
            store.removeModal(id);
            store.needCloseModals.delete(id);
            resolvePromise();
        }
        (_modalProps_onClosed = modalProps.onClosed) === null || _modalProps_onClosed === void 0 ? void 0 : _modalProps_onClosed.call(modalProps);
    };
    return {
        onClose,
        onClosed
    };
};
const createModalData = (item, id, callbacks)=>{
    const { onClose, onClosed } = callbacks;
    switch(item.type){
        case 'card':
            return {
                ...item,
                id,
                modalProps: {
                    ...item.modalProps,
                    id,
                    onClose,
                    onClosed
                }
            };
        case 'page':
            return {
                ...item,
                id,
                modalProps: {
                    ...item.modalProps,
                    id,
                    onClose,
                    onClosed
                }
            };
    }
};
const resolveProps = (props)=>{
    if ('component' in props) {
        return props;
    }
    return {
        component: props
    };
};
export const useModalActions = ({ store, saveHistory })=>{
    const saveHistoryRef = React.useRef(saveHistory);
    React.useEffect(()=>{
        saveHistoryRef.current = saveHistory;
    }, [
        saveHistory
    ]);
    const open = React.useCallback((item)=>{
        const modalProps = item.modalProps || {};
        const id = item.id;
        let resolvePromise;
        const promise = new Promise((resolve)=>{
            resolvePromise = resolve;
        });
        const callbacks = createModalCallbacks(id, modalProps, store, resolvePromise);
        const newModalData = createModalData(item, id, callbacks);
        if (!saveHistoryRef.current) {
            store.closePrevActiveIfNoHistory();
        }
        store.addModal(newModalData);
        return {
            id,
            close: ()=>store.closeModal(id),
            onClose: (resolve)=>{
                return promise.then(resolve);
            },
            update: item.update
        };
    }, [
        store
    ]);
    const update = React.useCallback((...args)=>{
        // Тип нужен только для улучшения типизации при вызове функции
        // eslint-disable-next-line @typescript-eslint/no-unused-vars
        const [id, _, props] = args;
        store.updateModalProps(id, props);
    }, [
        store
    ]);
    const close = React.useCallback((id)=>{
        store.closeModal(id);
    }, [
        store
    ]);
    const openCustomModalCard = React.useCallback((props)=>{
        const { id: idProp, component, baseProps: modalProps, additionalProps } = resolveProps(props);
        const id = idProp || randomUUID();
        return open({
            type: 'card',
            id,
            component,
            additionalProps,
            modalProps: modalProps,
            close: ()=>close(id),
            update: (newProps)=>update(id, 'card', newProps)
        });
    }, [
        close,
        open,
        update
    ]);
    const openCustomModalPage = React.useCallback((props)=>{
        const { id: idProp, component, baseProps: modalProps, additionalProps } = resolveProps(props);
        const id = idProp || randomUUID();
        return open({
            type: 'page',
            id,
            component,
            additionalProps,
            modalProps: modalProps,
            close: ()=>close(id),
            update: (newProps)=>update(id, 'page', newProps)
        });
    }, [
        close,
        open,
        update
    ]);
    const openModalPage = React.useCallback((props)=>{
        return openCustomModalPage({
            id: props.id,
            component: ModalPageWrapper,
            baseProps: props
        });
    }, [
        openCustomModalPage
    ]);
    const openModalCard = React.useCallback((props)=>{
        return openCustomModalCard({
            id: props.id,
            component: ModalCardWrapper,
            baseProps: props
        });
    }, [
        openCustomModalCard
    ]);
    const closeAll = React.useCallback(()=>{
        store.closeAll();
    }, [
        store
    ]);
    return {
        openModalPage,
        openModalCard,
        openCustomModalCard,
        openCustomModalPage,
        close,
        update,
        closeAll
    };
};
