// @vkontakte/vkui/dist/components/CustomSelectDropdown/CustomSelectDropdown.js
/* eslint-disable jsdoc/require-jsdoc */ import { jsx as _jsx } from "react/jsx-runtime";
import { classNames } from "@vkontakte/vkjs";
import { CustomScrollView } from "../CustomScrollView/CustomScrollView.js";
import { Popper } from "../Popper/Popper.js";
import { Spinner } from "../Spinner/Spinner.js";
export const CustomSelectDropdown = ({ children, targetRef, scrollBoxRef, placement = 'bottom', fetching, offsetDistance = 0, autoWidth = false, forcePortal = true, className, noMaxHeight = false, // CustomScrollView
overscrollBehavior, ...restProps })=>{
    return /*#__PURE__*/ _jsx(Popper, {
        targetRef: targetRef,
        offsetByMainAxis: offsetDistance,
        sameWidth: !autoWidth,
        placement: placement,
        className: classNames("vkuiCustomSelectDropdown__host", 'vkuiInternalCustomSelectDropdown', offsetDistance === 0 && (placement.includes('top') ? "vkuiCustomSelectDropdown__top" : "vkuiCustomSelectDropdown__bottom"), autoWidth && classNames("vkuiCustomSelectDropdown__wide", 'vkuiInternalCustomSelectDropdown--wide'), className),
        usePortal: forcePortal,
        autoUpdateOnTargetResize: true,
        flipMiddlewareFallbackAxisSideDirection: "none",
        ...restProps,
        children: /*#__PURE__*/ _jsx(CustomScrollView, {
            getRootRef: scrollBoxRef,
            className: noMaxHeight ? undefined : "vkuiCustomSelectDropdown__inWithMaxHeight",
            overscrollBehavior: overscrollBehavior,
            tabIndex: -1,
            "aria-busy": fetching,
            children: fetching ? /*#__PURE__*/ _jsx("div", {
                className: "vkuiCustomSelectDropdown__fetching",
                children: /*#__PURE__*/ _jsx(Spinner, {
                    size: "s"
                })
            }) : children
        })
    });
};
