// @vkontakte/vkui/dist/components/CardGrid/CardGrid.js
'use client';
import { jsx as _jsx } from "react/jsx-runtime";
import { classNames } from "@vkontakte/vkjs";
import { useAdaptivity } from "../../hooks/useAdaptivity.js";
import { ViewWidth } from "../../lib/adaptivity/index.js";
import { RootComponent } from "../RootComponent/RootComponent.js";
function getViewWidthClassName(viewWidth, legacySizeX) {
    // TODO [>=10]: #9015 Удалить это условие
    if (legacySizeX !== undefined && legacySizeX === 'compact') {
        return "vkuiCardGrid__viewWidthNone";
    }
    if (viewWidth === 'none') {
        return "vkuiCardGrid__viewWidthSmallTabletMinus";
    }
    if (viewWidth < ViewWidth.SMALL_TABLET) {
        return "vkuiCardGrid__viewWidthNone";
    }
    return;
}
const stylesSize = {
    s: 'vkuiInternalCardGrid--size-s',
    m: 'vkuiInternalCardGrid--size-m',
    l: 'vkuiInternalCardGrid--size-l'
};
/**
 * @see https://vkui.io/components/card-grid
 */ export const CardGrid = ({ size = 's', padding = false, Component = 'ul', ...restProps })=>{
    const { sizeX: legacySizeX, viewWidth = 'none' } = useAdaptivity();
    return /*#__PURE__*/ _jsx(RootComponent, {
        ...restProps,
        Component: Component,
        baseClassName: classNames("vkuiCardGrid__host", 'vkuiInternalCardGrid', padding && "vkuiCardGrid__padding", stylesSize[size], getViewWidthClassName(viewWidth, legacySizeX))
    });
};
