// @vkontakte/vkui/dist/components/View/View.js
'use client';
import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
import * as React from "react";
import { classNames } from "@vkontakte/vkjs";
import { usePlatform } from "../../hooks/usePlatform.js";
import { millisecondsInSecond } from "../../lib/date.js";
import { blurActiveElement, useDOM } from "../../lib/dom.js";
import { getNavId } from "../../lib/getNavId.js";
import { warnOnce } from "../../lib/warnOnce.js";
import { useScroll } from "../AppRoot/ScrollContext.js";
import { useConfigProvider } from "../ConfigProvider/ConfigProviderContext.js";
import { NavViewIdContext } from "../NavIdContext/NavIdContext.js";
import { NavTransitionProvider } from "../NavTransitionContext/NavTransitionContext.js";
import { NavTransitionDirectionProvider } from "../NavTransitionDirectionContext/NavTransitionDirectionContext.js";
import { useSplitCol } from "../SplitCol/SplitColContext.js";
import { Touch } from "../Touch/Touch.js";
import { useLayoutEffectCall } from "./useLayoutEffectCall.js";
import { getSwipeBackPredicates, hasHorizontalScrollableElementWithScrolledToLeft, swipeBackExcluded } from "./utils.js";
export const scrollsCache = new Map();
const warn = warnOnce('View');
/**
 * @see https://vkui.io/components/view
 */ export const View = ({ activePanel: activePanelProp, history, nav, onTransition, onSwipeBack, onSwipeBackStart, onSwipeBackCancel: onSwipeBackCancelProp, children, className, ...restProps })=>{
    const id = getNavId({
        nav,
        id: restProps.id
    });
    const [scrolls] = React.useState(()=>scrollsCache.get(id) || new Map());
    const layoutEffectCall = useLayoutEffectCall();
    React.useEffect(()=>()=>{
            if (id) {
                scrollsCache.set(id, scrolls);
            }
        });
    const panelNodes = React.useRef({});
    const { window, document } = useDOM();
    const scroll = useScroll();
    const configProvider = useConfigProvider();
    const splitCol = useSplitCol();
    const platform = usePlatform();
    const [animated, setAnimated] = React.useState(false);
    const [visiblePanels, setVisiblePanels] = React.useState([
        activePanelProp
    ]);
    const [activePanel, setActivePanel] = React.useState(activePanelProp);
    const [isBack, setIsBack] = React.useState(undefined);
    const [prevPanel, setPrevPanel] = React.useState(null);
    const [nextPanel, setNextPanel] = React.useState(null);
    const swipeBackPrevented = React.useRef(false);
    const [swipingBack, setSwipingBack] = React.useState(undefined);
    const [swipeBackStartX, setSwipeBackStartX] = React.useState(0);
    const [swipeBackShift, setSwipeBackShift] = React.useState(0);
    const [swipeBackNextPanel, setSwipeBackNextPanel] = React.useState(null);
    const [swipeBackPrevPanel, setSwipeBackPrevPanel] = React.useState(null);
    const [swipeBackResult, setSwipeBackResult] = React.useState(null);
    const [browserSwipe, setBrowserSwipe] = React.useState(false);
    const prevActivePanel = React.useRef(undefined);
    const prevSwipingBack = React.useRef(undefined);
    const prevBrowserSwipe = React.useRef(undefined);
    const prevSwipeBackResult = React.useRef(undefined);
    const prevSwipeBackShift = React.useRef(undefined);
    const prevSwipeBackPrevPanel = React.useRef(undefined);
    const prevOnTransition = React.useRef(undefined);
    const panels = React.Children.toArray(children).filter((panel)=>{
        const panelId = getNavId(panel.props, warn);
        return panelId !== undefined && visiblePanels.includes(panelId) || panelId === swipeBackPrevPanel || panelId === swipeBackNextPanel;
    });
    const disableAnimation = !configProvider.transitionMotionEnabled || !splitCol.animate || platform === 'vkcom';
    const iOSSwipeBackSimulationEnabled = !disableAnimation && platform === 'ios' && configProvider.isWebView && Boolean(onSwipeBack);
    const flushTransition = React.useCallback((prevPanel, isBackTransition)=>{
        if (isBackTransition) {
            scrolls.set(prevPanel, 0);
        }
        setPrevPanel(null);
        setNextPanel(null);
        setVisiblePanels([
            activePanelProp
        ]);
        setActivePanel(activePanelProp);
        setAnimated(false);
        setIsBack(isBackTransition);
        layoutEffectCall(()=>{
            scroll === null || scroll === void 0 ? void 0 : scroll.scrollTo(0, isBackTransition ? scrolls.get(activePanelProp) : 0);
            onTransition && onTransition({
                isBack: isBackTransition,
                from: prevPanel,
                to: activePanelProp
            });
        });
    }, [
        activePanelProp,
        layoutEffectCall,
        onTransition,
        scroll,
        scrolls
    ]);
    const handleAnimatedTargetAnimationEnd = ()=>{
        if (prevPanel !== null) {
            flushTransition(prevPanel, Boolean(isBack));
        }
    };
    const onSwipeBackSuccess = React.useCallback(()=>{
        onSwipeBack && onSwipeBack();
    }, [
        onSwipeBack
    ]);
    const onSwipeBackCancel = React.useCallback(()=>{
        onSwipeBackCancelProp && onSwipeBackCancelProp();
        setSwipeBackPrevPanel(null);
        setSwipeBackNextPanel(null);
        setSwipingBack(false);
        setSwipeBackResult(null);
        setSwipeBackStartX(0);
        setSwipeBackShift(0);
    }, [
        onSwipeBackCancelProp
    ]);
    const swipingBackTransitionEndHandler = React.useCallback(()=>{
        switch(swipeBackResult){
            case 'fail':
                onSwipeBackCancel();
                break;
            case 'success':
                onSwipeBackSuccess();
        }
    }, [
        onSwipeBackCancel,
        onSwipeBackSuccess,
        swipeBackResult
    ]);
    const handleTouchMoveXForNativeIOSSwipeBackOrSwipeNext = (event)=>{
        if (browserSwipe) {
            return;
        }
        const { swipeBackTriggered, viewportStartEdgeTouched, viewportEndEdgeTouched } = getSwipeBackPredicates(event.startX, event.shiftX, window.innerWidth);
        if ((viewportStartEdgeTouched || viewportEndEdgeTouched) && swipeBackTriggered) {
            setBrowserSwipe(true);
        }
    };
    const handleTouchMoveXForIOSSwipeBackSimulation = (event)=>{
        if (swipeBackPrevented.current || swipeBackExcluded(event)) {
            return;
        }
        const { swipedToOpposite, swipeBackTriggered, viewportStartEdgeTouched } = getSwipeBackPredicates(event.startX, event.shiftX, window.innerWidth);
        if (animated && swipeBackTriggered) {
            return;
        }
        if (!swipingBack && history && history.length > 1) {
            if (swipedToOpposite) {
                swipeBackPrevented.current = true;
                return;
            }
            if (!swipeBackTriggered) {
                return;
            }
            if (!viewportStartEdgeTouched && hasHorizontalScrollableElementWithScrolledToLeft(event.originalEvent.target)) {
                swipeBackPrevented.current = true;
                return;
            }
            // Начался свайп назад
            if (onSwipeBackStart) {
                const payload = onSwipeBackStart(activePanel);
                if (payload === 'prevent') {
                    swipeBackPrevented.current = true;
                    return;
                }
            }
            if (activePanel !== null) {
                // Note: вызываем закрытие клавиатуры. В iOS это нативное поведение при свайпе.
                blurActiveElement(document);
                scrolls.set(activePanel, scroll === null || scroll === void 0 ? void 0 : scroll.getScroll().y);
            }
            setSwipingBack(true);
            setSwipeBackStartX(event.startX);
            setSwipeBackPrevPanel(activePanel);
            setSwipeBackNextPanel(history.slice(-2)[0]);
        }
        if (swipingBack) {
            if (event.shiftX < 0) {
                setSwipeBackShift(0);
            } else if (event.shiftX > window.innerWidth - swipeBackStartX) {
                setSwipeBackShift(window.innerWidth);
            } else {
                setSwipeBackShift(event.shiftX);
            }
        }
    };
    const handleTouchEndForIOSSwipeBackSimulation = (event)=>{
        swipeBackPrevented.current = false;
        if (swipingBack) {
            var _window_innerWidth;
            const speed = swipeBackShift / event.duration * millisecondsInSecond;
            if (swipeBackShift === 0) {
                onSwipeBackCancel();
            } else if (swipeBackShift >= ((_window_innerWidth = window.innerWidth) !== null && _window_innerWidth !== void 0 ? _window_innerWidth : 0)) {
                onSwipeBackSuccess();
            } else if (speed > 250 || swipeBackShift >= window.innerWidth / 2) {
                setSwipeBackResult('success');
            } else {
                setSwipeBackResult('fail');
            }
        }
    };
    const calcPanelSwipeStyles = (isPrev, isNext)=>{
        if (!isPrev && !isNext || swipeBackResult) {
            return {};
        }
        if (isNext) {
            return window ? {
                transform: `translate3d(${-50 + swipeBackShift * 100 / window.innerWidth / 2}%, 0, 0)`
            } : {};
        }
        if (isPrev) {
            return {
                transform: `translate3d(${swipeBackShift}px, 0, 0)`
            };
        }
        return {};
    };
    const calcPanelSwipeBackOverlayStyles = (isNext)=>{
        if (!window || !isNext) {
            return {};
        }
        const opacityOnSwipeEnd = swipeBackResult === 'success' ? 0 : swipeBackResult === 'fail' ? 1 : null;
        return {
            display: 'block',
            opacity: opacityOnSwipeEnd === null ? 1 - swipeBackShift / window.innerWidth : opacityOnSwipeEnd
        };
    };
    const handleSwipeBackTargetTransitionEnd = (event)=>{
        if (event.propertyName.includes('transform')) {
            swipingBackTransitionEndHandler();
        }
    };
    React.useEffect(()=>{
        // Нужен переход
        if (prevActivePanel.current && prevActivePanel.current !== activePanelProp && !prevSwipingBack.current && !prevBrowserSwipe.current) {
            const firstLayerId = React.Children.toArray(children).map((panel)=>getNavId(panel.props, warn)).find((id)=>id === prevActivePanel.current || id === activePanelProp);
            const isBackTransition = firstLayerId === activePanelProp;
            scrolls.set(prevActivePanel.current, scroll === null || scroll === void 0 ? void 0 : scroll.getScroll({
                compensateKeyboardHeight: false
            }).y);
            if (disableAnimation) {
                flushTransition(prevActivePanel.current, isBackTransition);
            } else {
                blurActiveElement(document);
                setVisiblePanels([
                    prevActivePanel.current,
                    activePanelProp
                ]);
                setPrevPanel(prevActivePanel.current);
                setNextPanel(activePanelProp);
                setActivePanel(null);
                setAnimated(true);
                setIsBack(isBackTransition);
            }
        }
        // Закончилась анимация свайпа назад
        if (prevActivePanel.current && prevActivePanel.current !== activePanelProp && prevSwipingBack.current) {
            const nextPanel = activePanelProp;
            const prevPanel = prevActivePanel.current;
            if (prevSwipeBackPrevPanel.current) {
                scrolls.set(prevSwipeBackPrevPanel.current, 0);
            }
            setSwipeBackPrevPanel(null);
            setSwipeBackNextPanel(null);
            setSwipingBack(false);
            setSwipeBackResult(null);
            setSwipeBackStartX(0);
            setSwipeBackShift(0);
            setActivePanel(nextPanel);
            setVisiblePanels([
                nextPanel
            ]);
            setIsBack(true);
            layoutEffectCall(()=>{
                if (nextPanel !== null) {
                    scroll === null || scroll === void 0 ? void 0 : scroll.scrollTo(0, scrolls.get(nextPanel));
                }
                prevOnTransition.current && prevOnTransition.current({
                    isBack: true,
                    from: prevPanel,
                    to: nextPanel
                });
            });
        }
        // Началась анимация завершения свайпа назад.
        // см. `onTransitionEnd()`
        // Закончился Safari свайп
        if (prevActivePanel.current !== activePanelProp && browserSwipe) {
            setBrowserSwipe(false);
            setNextPanel(null);
            setPrevPanel(null);
            setAnimated(false);
            setVisiblePanels([
                activePanelProp
            ]);
            setActivePanel(activePanelProp);
        }
    }, [
        activePanelProp,
        activePanel,
        browserSwipe,
        children,
        disableAnimation,
        document,
        flushTransition,
        scroll,
        swipeBackNextPanel,
        swipeBackResult,
        layoutEffectCall,
        scrolls
    ]);
    React.useEffect(function restoreScrollPositionWhenSwipeBackIsCancelled() {
        // Если свайп назад отменился (когда пользователь недостаточно сильно свайпнул)
        const swipeBackCancelledInTheMiddleOfAction = prevSwipeBackResult.current === 'fail' && !swipeBackResult;
        const swipeBackCancelledByMovingPanelBackToInitialPoint = prevSwipingBack.current && !swipingBack && prevSwipeBackShift.current === 0;
        if ((swipeBackCancelledInTheMiddleOfAction || swipeBackCancelledByMovingPanelBackToInitialPoint) && activePanel !== null) {
            scroll === null || scroll === void 0 ? void 0 : scroll.scrollTo(0, scrolls.get(activePanel));
        }
    }, [
        swipeBackResult,
        swipingBack,
        activePanel,
        scroll,
        scrolls
    ]);
    /**
   * @see https://github.com/VKCOM/VKUI/pull/9274
   */ React.useEffect(function updateAfterAll() {
        prevActivePanel.current = activePanelProp;
        prevSwipingBack.current = swipingBack;
        prevBrowserSwipe.current = browserSwipe;
        prevSwipeBackResult.current = swipeBackResult;
        prevSwipeBackShift.current = swipeBackShift;
        prevSwipeBackPrevPanel.current = swipeBackPrevPanel;
        prevOnTransition.current = onTransition;
    });
    return /*#__PURE__*/ _jsx(NavViewIdContext.Provider, {
        value: id,
        children: /*#__PURE__*/ _jsx(Touch, {
            Component: "section",
            ...restProps,
            className: classNames("vkuiView__host", platform === 'ios' && classNames("vkuiView__ios", 'vkuiInternalView--ios'), !disableAnimation && animated && "vkuiView__animated", !disableAnimation && swipingBack && "vkuiView__swipingBack", disableAnimation && "vkuiView__noMotion", className),
            onMoveX: iOSSwipeBackSimulationEnabled ? handleTouchMoveXForIOSSwipeBackSimulation : platform === 'ios' ? handleTouchMoveXForNativeIOSSwipeBackOrSwipeNext : undefined,
            onEnd: iOSSwipeBackSimulationEnabled ? handleTouchEndForIOSSwipeBackSimulation : undefined,
            children: /*#__PURE__*/ _jsx("div", {
                className: "vkuiView__panels",
                children: panels.map((panel)=>{
                    const panelId = getNavId(panel.props, warn);
                    const isPanelActive = panelId === activePanel;
                    const isPanelPrev = panelId === prevPanel;
                    const isPanelNext = panelId === nextPanel;
                    const isAnimatedTarget = animated && (isBack ? isPanelPrev : isPanelNext);
                    const isSwipeBackPrev = panelId === swipeBackPrevPanel;
                    const isSwipeBackNext = panelId === swipeBackNextPanel;
                    const isSwipeBackTarget = swipeBackResult && isSwipeBackPrev;
                    let scrollCompensateStyle = undefined;
                    if (isPanelPrev || isPanelNext && isBack || isSwipeBackPrev || isSwipeBackNext) {
                        const marginTop = scrolls.get(panelId);
                        if (marginTop !== undefined) {
                            scrollCompensateStyle = {
                                marginTop: -1 * marginTop
                            };
                        }
                    }
                    return /*#__PURE__*/ _jsxs("div", {
                        className: classNames("vkuiView__panel", isPanelActive && "vkuiView__panelActive", isPanelPrev && "vkuiView__panelPrev", isPanelNext && "vkuiView__panelNext", isSwipeBackPrev && "vkuiView__panelSwipeBackPrev", isSwipeBackNext && "vkuiView__panelSwipeBackNext", swipeBackResult === 'success' && "vkuiView__panelSwipeBackSuccess", swipeBackResult === 'fail' && "vkuiView__panelSwipeBackFailed"),
                        onTransitionEnd: isSwipeBackTarget ? handleSwipeBackTargetTransitionEnd : undefined,
                        onAnimationEnd: isAnimatedTarget ? handleAnimatedTargetAnimationEnd : undefined,
                        ref: (el)=>{
                            panelId !== undefined && (panelNodes.current[panelId] = el);
                        },
                        style: calcPanelSwipeStyles(isSwipeBackPrev, isSwipeBackNext),
                        children: [
                            platform === 'ios' && /*#__PURE__*/ _jsx("div", {
                                className: "vkuiView__panelOverlay",
                                style: calcPanelSwipeBackOverlayStyles(isSwipeBackNext)
                            }),
                            /*#__PURE__*/ _jsx("div", {
                                className: "vkuiView__panelIn",
                                style: scrollCompensateStyle,
                                children: /*#__PURE__*/ _jsx(NavTransitionDirectionProvider, {
                                    isBack: swipingBack || isBack,
                                    children: /*#__PURE__*/ _jsx(NavTransitionProvider, {
                                        entering: panelId === nextPanel || panelId === swipeBackNextPanel,
                                        children: panel
                                    })
                                })
                            })
                        ]
                    }, panelId);
                })
            })
        })
    });
};
