// @vkontakte/vkui/dist/cssm/components/Typography/Footnote/Footnote.js
'use client';
import { jsx as _jsx } from "react/jsx-runtime";
import { classNames } from "@vkontakte/vkjs";
import { useAdaptivity } from "../../../hooks/useAdaptivity.js";
import { Typography } from "../Typography.js";
import styles from "./Footnote.module.css";
const densityClassNames = {
    none: styles.densityNone,
    compact: styles.densityCompact
};
export function footnoteClassNames(density, caps = false) {
    return classNames(styles.host, density !== 'regular' && densityClassNames[density], caps && styles.caps);
}
/**
 * Используется для основных подписей.
 *
 * @see https://vkui.io/components/typography#footnote
 */ export const Footnote = ({ className, caps, Component = 'span', normalize = true, inline = false, ...restProps })=>{
    const { density = 'none' } = useAdaptivity();
    return /*#__PURE__*/ _jsx(Typography, {
        Component: Component,
        normalize: normalize,
        inline: inline,
        className: classNames(className, footnoteClassNames(density, caps)),
        ...restProps
    });
};
