// @vkontakte/vkui/dist/components/ActionSheet/ActionSheet.js
'use client';
import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
import * as React from "react";
import { noop } from "@vkontakte/vkjs";
import { useAdaptivityWithJSMediaQueries } from "../../hooks/useAdaptivityWithJSMediaQueries.js";
import { usePlatform } from "../../hooks/usePlatform.js";
import { useStableCallback } from "../../hooks/useStableCallback.js";
import { useCSSKeyframesAnimationController } from "../../lib/animation/index.js";
import { AppRootPortal } from "../AppRoot/AppRootPortal.js";
import { useScrollLock } from "../AppRoot/ScrollContext.js";
import { PopoutWrapper } from "../PopoutWrapper/PopoutWrapper.js";
import { Footnote } from "../Typography/Footnote/Footnote.js";
import { ActionSheetContext } from "./ActionSheetContext.js";
import { ActionSheetDefaultIosCloseItem } from "./ActionSheetDefaultIosCloseItem.js";
import { ActionSheetDropdownMenu } from "./ActionSheetDropdownMenu.js";
import { ActionSheetDropdownSheet } from "./ActionSheetDropdownSheet.js";
/**
 * @see https://vkui.io/components/action-sheet
 */ export const ActionSheet = ({ children, className, title, description, style, iosCloseItem, slotProps, popupOffsetDistance, placement, mode: modeProp, onClose: onCloseProp, onClosed, ...restProps })=>{
    const platform = usePlatform();
    const [closedReason, setClosedReason] = React.useState(undefined);
    const onCloseStable = useStableCallback(onCloseProp || noop);
    const onClose = React.useCallback((onCloseReason, onClosedReason)=>{
        onCloseStable(onCloseReason);
        setClosedReason(onClosedReason);
    }, [
        onCloseStable
    ]);
    const onOverlayClick = React.useCallback(()=>onClose('click-overlay', 'other'), [
        onClose
    ]);
    const actionCallbackRef = React.useRef(noop);
    const [animationState, animationHandlers] = useCSSKeyframesAnimationController(closedReason !== undefined ? 'exit' : 'enter', {
        onExited () {
            onClosed({
                closedBy: closedReason || 'other'
            });
            actionCallbackRef.current();
            actionCallbackRef.current = noop;
        }
    });
    const { isDesktop } = useAdaptivityWithJSMediaQueries();
    const mode = modeProp !== null && modeProp !== void 0 ? modeProp : isDesktop ? 'menu' : 'sheet';
    useScrollLock(mode === 'sheet');
    const onItemClick = React.useCallback(({ action, immediateAction, autoClose, isCancelItem })=>(event)=>{
            event.persist();
            immediateAction && immediateAction(event);
            if (autoClose) {
                if (action) {
                    actionCallbackRef.current = ()=>action(event);
                }
                onClose(isCancelItem ? 'click-cancel-item' : 'click-action-item', isCancelItem ? 'cancel-item' : 'action-item');
            } else {
                action && action(event);
            }
        }, [
        onClose
    ]);
    const contextValue = React.useMemo(()=>({
            onItemClick,
            mode,
            onClose: (reason)=>onClose(reason, 'other')
        }), [
        mode,
        onClose,
        onItemClick
    ]);
    const DropdownComponent = mode === 'menu' ? ActionSheetDropdownMenu : ActionSheetDropdownSheet;
    const dropdownProps = mode === 'menu' ? Object.assign(restProps, {
        popupOffsetDistance,
        placement
    }) : restProps;
    const actionSheet = /*#__PURE__*/ _jsx(ActionSheetContext.Provider, {
        value: contextValue,
        children: /*#__PURE__*/ _jsxs(DropdownComponent, {
            closing: Boolean(closedReason),
            role: "dialog",
            "aria-modal": "true",
            autoFocus: animationState === 'entered',
            ...dropdownProps,
            ...animationHandlers,
            className: mode === 'menu' ? className : undefined,
            style: mode === 'menu' ? style : undefined,
            children: [
                /*#__PURE__*/ _jsxs("div", {
                    className: "vkuiActionSheet__contentWrapper",
                    children: [
                        (title || description) && /*#__PURE__*/ _jsxs("div", {
                            className: "vkuiActionSheet__header",
                            children: [
                                title && /*#__PURE__*/ _jsx(Footnote, {
                                    weight: "2",
                                    className: "vkuiActionSheet__title",
                                    children: title
                                }),
                                description && /*#__PURE__*/ _jsx(Footnote, {
                                    className: "vkuiActionSheet__description",
                                    children: description
                                })
                            ]
                        }),
                        children
                    ]
                }),
                platform === 'ios' && mode === 'sheet' && /*#__PURE__*/ _jsx("div", {
                    className: "vkuiActionSheet__closeItemWrapperIos",
                    children: iosCloseItem !== null && iosCloseItem !== void 0 ? iosCloseItem : /*#__PURE__*/ _jsx(ActionSheetDefaultIosCloseItem, {
                        ...slotProps === null || slotProps === void 0 ? void 0 : slotProps.iosCloseItem
                    })
                })
            ]
        })
    });
    return /*#__PURE__*/ _jsx(AppRootPortal, {
        children: /*#__PURE__*/ _jsx(PopoutWrapper, {
            noBackground: mode === 'menu',
            closing: Boolean(closedReason),
            alignY: "bottom",
            className: className,
            style: style,
            onClick: onOverlayClick,
            children: actionSheet
        })
    });
};
