// @vkontakte/vkui/dist/components/ActionSheet/ActionSheetDropdownSheet.js
'use client';
import { jsx as _jsx } from "react/jsx-runtime";
import * as React from "react";
import { classNames } from "@vkontakte/vkjs";
import { useAdaptivityWithJSMediaQueries } from "../../hooks/useAdaptivityWithJSMediaQueries.js";
import { useExternRef } from "../../hooks/useExternRef.js";
import { useGlobalEscKeyDown } from "../../hooks/useGlobalEscKeyDown.js";
import { usePlatform } from "../../hooks/usePlatform.js";
import { stopPropagation } from "../../lib/utils.js";
import { FocusTrap } from "../FocusTrap/FocusTrap.js";
import { RootComponent } from "../RootComponent/RootComponent.js";
import { ActionSheetContext } from "./ActionSheetContext.js";
export const ActionSheetDropdownSheet = ({ getRootRef, children, closing, // these 2 props are only omitted - ActionSheetDesktop compat
toggleRef, className, onClick, allowClickPropagation = false, onClose: onCloseProp, // FocusTrap props
autoFocus, restoreFocus, disabled, timeout, ...restProps })=>{
    const { density } = useAdaptivityWithJSMediaQueries();
    const focusTrapRootRef = useExternRef(getRootRef);
    const platform = usePlatform();
    const { onClose: onActionSheetClose } = React.useContext(ActionSheetContext);
    const onClose = React.useCallback(()=>{
        onCloseProp === null || onCloseProp === void 0 ? void 0 : onCloseProp();
        onActionSheetClose === null || onActionSheetClose === void 0 ? void 0 : onActionSheetClose('escape-key');
    }, [
        onActionSheetClose,
        onCloseProp
    ]);
    const handleClick = allowClickPropagation ? onClick : (event)=>{
        stopPropagation(event);
        onClick === null || onClick === void 0 ? void 0 : onClick(event);
    };
    useGlobalEscKeyDown(true, onClose);
    return /*#__PURE__*/ _jsx(FocusTrap, {
        autoFocus: autoFocus,
        autoFocusDelay: timeout,
        restoreFocus: restoreFocus,
        disabled: disabled,
        rootRef: focusTrapRootRef,
        children: /*#__PURE__*/ _jsx(RootComponent, {
            ...restProps,
            tabIndex: -1,
            getRootRef: focusTrapRootRef,
            onClick: handleClick,
            className: classNames("vkuiActionSheet__host", platform === 'ios' && "vkuiActionSheet__ios", closing ? "vkuiActionSheet__closing" : "vkuiActionSheet__opening", density === 'compact' && "vkuiActionSheet__densityCompact", className),
            children: children
        })
    });
};
