// @vkontakte/vkui/dist/cssm/components/PanelHeaderContext/PanelHeaderContext.js
'use client';
import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
import * as React from "react";
import { classNames } from "@vkontakte/vkjs";
import { useAdaptivity } from "../../hooks/useAdaptivity.js";
import { useGlobalOnClickOutside } from "../../hooks/useGlobalOnClickOutside.js";
import { usePlatform } from "../../hooks/usePlatform.js";
import { ViewWidth } from "../../lib/adaptivity/index.js";
import { useCSSKeyframesAnimationController } from "../../lib/animation/index.js";
import { useScrollLock } from "../AppRoot/ScrollContext.js";
import { FixedLayout } from "../FixedLayout/FixedLayout.js";
import styles from "./PanelHeaderContext.module.css";
function getViewWidthClassName(viewWidth, legacySizeX) {
    // TODO [>=10]: #9015 Удалить это условие
    if (legacySizeX !== undefined) {
        return legacySizeX === 'regular' ? styles.viewWidthSmallTabletPlus : styles.viewWidthSmallTabletMinus;
    }
    if (viewWidth === 'none') {
        return classNames(styles.viewWidthNone, 'vkuiInternalGroup--viewWidth-none');
    }
    return viewWidth >= ViewWidth.SMALL_TABLET ? styles.viewWidthSmallTabletPlus : styles.viewWidthSmallTabletMinus;
}
/**
 * @see https://vkui.io/components/panel-header-context
 */ export const PanelHeaderContext = ({ children, opened = false, className, onClose, ...restProps })=>{
    const platform = usePlatform();
    const { sizeX: legacySizeX, viewWidth = 'none' } = useAdaptivity();
    const elementRef = React.useRef(null);
    const [animationState, animationHandlers] = useCSSKeyframesAnimationController(opened ? 'enter' : 'exit', undefined, true);
    const visible = animationState !== 'exited';
    useScrollLock(platform !== 'vkcom' && visible);
    const handleGlobalOnClickOutside = React.useCallback((event)=>{
        if (opened) {
            event.stopPropagation();
            onClose();
        }
    }, [
        opened,
        onClose
    ]);
    useGlobalOnClickOutside(handleGlobalOnClickOutside, visible ? elementRef : null);
    if (!visible) {
        return null;
    }
    return /*#__PURE__*/ _jsxs(FixedLayout, {
        ...restProps,
        className: classNames(styles.host, platform === 'ios' && styles.ios, opened ? styles.opened : styles.closing, getViewWidthClassName(viewWidth, legacySizeX), className),
        vertical: "top",
        children: [
            /*#__PURE__*/ _jsx("div", {
                onClick: (event)=>{
                    event.stopPropagation();
                    onClose();
                },
                className: styles.fade
            }),
            /*#__PURE__*/ _jsx("div", {
                "data-testid": process.env.NODE_ENV === 'test' ? 'content' : undefined,
                className: styles.in,
                ref: elementRef,
                ...animationHandlers,
                children: /*#__PURE__*/ _jsx("div", {
                    className: styles.content,
                    children: children
                })
            })
        ]
    });
};
