// @vkontakte/vkui/dist/cssm/components/Panel/Panel.js
'use client';
import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
import * as React from "react";
import { classNames } from "@vkontakte/vkjs";
import { useAdaptivity } from "../../hooks/useAdaptivity.js";
import { ViewWidth } from "../../lib/adaptivity/index.js";
import { AppRootContext } from "../AppRoot/AppRootContext.js";
import { NavPanelIdContext } from "../NavIdContext/NavIdContext.js";
import { OnboardingTooltipContainer } from "../OnboardingTooltip/OnboardingTooltipContainer.js";
import { RootComponent } from "../RootComponent/RootComponent.js";
import { Touch } from "../Touch/Touch.js";
import styles from "./Panel.module.css";
function getViewWidthClassName(viewWidth, legacySizeX) {
    // TODO [>=10]: #9015 Удалить это условие
    if (legacySizeX !== undefined) {
        return legacySizeX === 'regular' ? styles.viewWidthSmallTabletPlus : styles.viewWidthSmallTabletMinus;
    }
    if (viewWidth === 'none') {
        return classNames(styles.viewWidthNone, 'vkuiInternalGroup--viewWidth-none');
    }
    return viewWidth >= ViewWidth.SMALL_TABLET ? styles.viewWidthSmallTabletPlus : styles.viewWidthSmallTabletMinus;
}
const stylesMode = {
    none: styles.modeNone,
    plain: styles.modePlain,
    card: styles.modeCard
};
/**
 * @see https://vkui.io/components/panel
 */ export const Panel = ({ centered = false, children, nav, mode: modeProp, disableBackground, ...restProps })=>{
    const { sizeX: legacySizeX, viewWidth = 'none' } = useAdaptivity();
    const mode = usePanelMode(modeProp, viewWidth, legacySizeX);
    return /*#__PURE__*/ _jsx(NavPanelIdContext.Provider, {
        value: restProps.id || nav,
        children: /*#__PURE__*/ _jsx(RootComponent, {
            ...restProps,
            baseClassName: classNames(styles.host, getViewWidthClassName(viewWidth, legacySizeX), centered && 'vkuiInternalPanel--centered', disableBackground && styles.disableBackground, stylesMode[mode]),
            children: /*#__PURE__*/ _jsxs(Touch, {
                Component: OnboardingTooltipContainer,
                className: classNames(styles.in, 'vkuiInternalPanel__in'),
                children: [
                    /*#__PURE__*/ _jsx("div", {
                        className: styles.inBefore
                    }),
                    centered ? /*#__PURE__*/ _jsx("div", {
                        className: styles.centered,
                        children: children
                    }) : children,
                    /*#__PURE__*/ _jsx("div", {
                        className: styles.inAfter
                    })
                ]
            })
        })
    });
};
function usePanelMode(modeProp, viewWidth, legacySizeX) {
    const { layout } = React.useContext(AppRootContext);
    if (modeProp) {
        return modeProp;
    }
    if (layout) {
        return layout;
    }
    // TODO [>=10]: #9015 Удалить это условие
    if (legacySizeX !== undefined) {
        return legacySizeX === 'regular' ? 'card' : 'plain';
    }
    if (viewWidth !== 'none') {
        return viewWidth >= ViewWidth.SMALL_TABLET ? 'card' : 'plain';
    }
    return 'none';
}
