// @vkontakte/vkui/dist/components/Textarea/Textarea.js
'use client';
import { jsx as _jsx } from "react/jsx-runtime";
import * as React from "react";
import { classNames } from "@vkontakte/vkjs";
import { useAdaptivity } from "../../hooks/useAdaptivity.js";
import { useExternRef } from "../../hooks/useExternRef.js";
import { useMergeProps } from "../../hooks/useMergeProps.js";
import { usePlatform } from "../../hooks/usePlatform.js";
import { useResizeObserver } from "../../hooks/useResizeObserver.js";
import { callMultiple } from "../../lib/callMultiple.js";
import { useDOM } from "../../lib/dom.js";
import { warnOnce } from "../../lib/warnOnce.js";
import { FormField } from "../FormField/FormField.js";
import { UnstyledTextField } from "../UnstyledTextField/UnstyledTextField.js";
import { useResizeTextarea } from "./useResizeTextarea.js";
const warn = warnOnce('Textarea');
const densityClassNames = {
    none: "vkuiTextarea__densityNone",
    compact: "vkuiTextarea__densityCompact"
};
/**
 * @see https://vkui.io/components/textarea
 */ export const Textarea = ({ // FormFieldProps
status, mode, after, before, afterAlign, beforeAlign, maxHeight, // TextareaProps
grow = true, onResize, align, getRef, // textarea props
autoComplete, autoCapitalize, autoCorrect, cols, dirName, disabled, maxLength, minLength, name, placeholder, readOnly, required, value: valueProp, wrap, rows = 2, form, onChange: onChangeProp, onFocus, onBlur, id, inputMode, defaultValue, autoFocus, tabIndex, spellCheck, slotProps, ...restProps })=>{
    /* istanbul ignore if: не проверяем в тестах */ if (process.env.NODE_ENV === 'development' && getRef) {
        warn('Свойство `getRef` устаревшее, используйте `slotProps={ textArea: { getRootRef: ... } }`');
    }
    const { density = 'none' } = useAdaptivity();
    const platform = usePlatform();
    const { window } = useDOM();
    const { className, ...rootProps } = useMergeProps(restProps, slotProps === null || slotProps === void 0 ? void 0 : slotProps.root);
    const { onChange, getRootRef: getTextAreaRef, value, ...textAreaRest } = useMergeProps({
        className: "vkuiTextarea__el",
        getRootRef: getRef,
        autoComplete,
        autoCapitalize,
        autoCorrect,
        cols,
        dirName,
        disabled,
        maxLength,
        minLength,
        name,
        placeholder,
        readOnly,
        required,
        value: valueProp,
        wrap,
        rows,
        form,
        onChange: onChangeProp,
        onFocus,
        onBlur,
        id,
        inputMode,
        defaultValue,
        autoFocus,
        tabIndex,
        spellCheck
    }, slotProps === null || slotProps === void 0 ? void 0 : slotProps.textArea);
    const [refResizeTextarea, resize] = useResizeTextarea(onResize, grow);
    const elementRef = useExternRef(getTextAreaRef, refResizeTextarea);
    React.useEffect(resize, [
        resize,
        density,
        platform,
        value
    ]);
    useResizeObserver(window, resize);
    return /*#__PURE__*/ _jsx(FormField, {
        className: classNames("vkuiTextarea__host", density !== 'regular' && densityClassNames[density], align === 'right' && "vkuiTextarea__alignRight", align === 'center' && "vkuiTextarea__alignCenter", className),
        disabled: textAreaRest.disabled,
        status: status,
        mode: mode,
        after: after,
        before: before,
        afterAlign: afterAlign,
        beforeAlign: beforeAlign,
        maxHeight: maxHeight,
        ...rootProps,
        children: /*#__PURE__*/ _jsx(UnstyledTextField, {
            value: value,
            as: "textarea",
            rows: rows,
            getRootRef: elementRef,
            onChange: callMultiple(onChange, resize),
            ...textAreaRest
        })
    });
};
