// @vkontakte/vkui/dist/components/Checkbox/CheckboxInput/CheckboxInput.js
'use client';
import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
import * as React from "react";
import { Icon20CheckBoxIndetermanate, Icon20CheckBoxOff, Icon20CheckBoxOn, Icon24CheckBoxOff, Icon24CheckBoxOn } from "@vkontakte/icons";
import { classNames } from "@vkontakte/vkjs";
import { useAdaptivityConditionalRender } from "../../../hooks/useAdaptivityConditionalRender/index.js";
import { useExternRef } from "../../../hooks/useExternRef.js";
import { useMergeProps } from "../../../hooks/useMergeProps.js";
import { usePlatform } from "../../../hooks/usePlatform.js";
import { warnOnce } from "../../../lib/warnOnce.js";
import { RootComponent } from "../../RootComponent/RootComponent.js";
import { VisuallyHidden } from "../../VisuallyHidden/VisuallyHidden.js";
function setIndeterminate(el, indeterminate) {
    el.indeterminate = indeterminate;
}
const warn = warnOnce('Checkbox');
export function CheckboxInput({ getRef, // Input props
checked, defaultChecked, disabled, readOnly, required, autoFocus, id, name, value, onChange: onChangeProp, onFocus, onBlur, // CheckboxInputProps
indeterminate, defaultIndeterminate, IconOnCompact = Icon20CheckBoxOn, IconOnRegular = Icon24CheckBoxOn, IconOffCompact = Icon20CheckBoxOff, IconOffRegular = Icon24CheckBoxOff, IconIndeterminate = Icon20CheckBoxIndetermanate, slotProps, ...restProps }) {
    const rootRest = useMergeProps(restProps, slotProps === null || slotProps === void 0 ? void 0 : slotProps.root);
    const { onChange, getRootRef: getInputRef, className: inputClassName, ...inputRest } = useMergeProps({
        getRootRef: getRef,
        checked,
        defaultChecked,
        disabled,
        readOnly,
        required,
        autoFocus,
        id,
        name,
        value,
        onChange: onChangeProp,
        onFocus,
        onBlur
    }, slotProps === null || slotProps === void 0 ? void 0 : slotProps.input);
    const inputRef = useExternRef(getInputRef);
    const platform = usePlatform();
    const { density: adaptiveDensity } = useAdaptivityConditionalRender();
    React.useEffect(()=>{
        const indeterminateValue = indeterminate === undefined ? defaultIndeterminate : indeterminate;
        if (inputRef.current) {
            setIndeterminate(inputRef.current, Boolean(indeterminateValue));
        }
    }, [
        defaultIndeterminate,
        indeterminate,
        inputRef
    ]);
    const handleChange = React.useCallback((event)=>{
        if (defaultIndeterminate !== undefined && indeterminate === undefined && inputRest.checked === undefined && inputRef.current) {
            setIndeterminate(inputRef.current, false);
        }
        if (indeterminate !== undefined && inputRef.current) {
            setIndeterminate(inputRef.current, Boolean(indeterminate));
        }
        onChange && onChange(event);
    }, [
        defaultIndeterminate,
        indeterminate,
        inputRest.checked,
        onChange,
        inputRef
    ]);
    if (process.env.NODE_ENV === 'development') {
        /* istanbul ignore if: не проверяем в тестах */ if (getRef) {
            warn('Свойство `getRef` устаревшее, используйте `slotProps={ input: { getRootRef: ... } }`');
        }
        if (defaultIndeterminate && inputRest.defaultChecked) {
            warn('defaultIndeterminate и defaultChecked не могут быть true одновременно', 'error');
        }
        if (indeterminate && inputRest.checked) {
            warn('indeterminate и checked не могут быть true одновременно', 'error');
        }
        if (inputRest.defaultChecked && inputRest.checked) {
            warn('defaultChecked и checked не могут быть true одновременно', 'error');
        }
    }
    return /*#__PURE__*/ _jsxs(RootComponent, {
        baseClassName: "vkuiCheckboxInput__host",
        ...rootRest,
        children: [
            /*#__PURE__*/ _jsx(VisuallyHidden, {
                Component: "input",
                type: "checkbox",
                onChange: handleChange,
                getRootRef: inputRef,
                className: classNames(inputClassName, "vkuiCheckboxInput__input"),
                ...inputRest
            }),
            platform === 'vkcom' ? /*#__PURE__*/ _jsx(IconOnCompact, {
                className: "vkuiCheckboxInput__iconOn"
            }) : /*#__PURE__*/ _jsxs(React.Fragment, {
                children: [
                    adaptiveDensity.compact && /*#__PURE__*/ _jsx(IconOnCompact, {
                        className: classNames("vkuiCheckboxInput__iconOn", adaptiveDensity.compact.className)
                    }),
                    adaptiveDensity.regular && /*#__PURE__*/ _jsx(IconOnRegular, {
                        className: classNames("vkuiCheckboxInput__iconOn", adaptiveDensity.regular.className)
                    })
                ]
            }),
            platform === 'vkcom' ? /*#__PURE__*/ _jsx(IconOffCompact, {
                className: "vkuiCheckboxInput__iconOff"
            }) : /*#__PURE__*/ _jsxs(React.Fragment, {
                children: [
                    adaptiveDensity.compact && /*#__PURE__*/ _jsx(IconOffCompact, {
                        className: classNames("vkuiCheckboxInput__iconOff", adaptiveDensity.compact.className)
                    }),
                    adaptiveDensity.regular && /*#__PURE__*/ _jsx(IconOffRegular, {
                        className: classNames("vkuiCheckboxInput__iconOff", adaptiveDensity.regular.className)
                    })
                ]
            }),
            platform === 'vkcom' ? /*#__PURE__*/ _jsx(IconIndeterminate, {
                width: 20,
                height: 20,
                className: "vkuiCheckboxInput__iconIndeterminate"
            }) : /*#__PURE__*/ _jsxs(React.Fragment, {
                children: [
                    adaptiveDensity.compact && /*#__PURE__*/ _jsx(IconIndeterminate, {
                        className: classNames("vkuiCheckboxInput__iconIndeterminate", adaptiveDensity.compact.className),
                        width: 20,
                        height: 20
                    }),
                    adaptiveDensity.regular && /*#__PURE__*/ _jsx(IconIndeterminate, {
                        className: classNames("vkuiCheckboxInput__iconIndeterminate", adaptiveDensity.regular.className),
                        width: 24,
                        height: 24
                    })
                ]
            })
        ]
    });
}
