// @vkontakte/vkui/dist/components/Group/GroupContainer.js
'use client';
import { jsx as _jsx, jsxs as _jsxs, Fragment as _Fragment } from "react/jsx-runtime";
import * as React from "react";
import { classNames } from "@vkontakte/vkjs";
import { useModalContext } from "../../context/ModalContext.js";
import { useAdaptivity } from "../../hooks/useAdaptivity.js";
import { ViewWidth } from "../../lib/adaptivity/index.js";
import { warnOnce } from "../../lib/warnOnce.js";
import { AppRootContext } from "../AppRoot/AppRootContext.js";
import { RootComponent } from "../RootComponent/RootComponent.js";
function getViewWidthClassName(viewWidth, legacySizeX) {
    // TODO [>=10]: #9015 Удалить это условие
    if (legacySizeX !== undefined) {
        return legacySizeX === 'regular' ? "vkuiGroup__viewWidthSmallTabletPlus" : "vkuiGroup__viewWidthSmallTabletMinus";
    }
    if (viewWidth === 'none') {
        return classNames("vkuiGroup__viewWidthNone", 'vkuiInternalGroup--viewWidth-none');
    }
    return viewWidth >= ViewWidth.SMALL_TABLET ? "vkuiGroup__viewWidthSmallTabletPlus" : "vkuiGroup__viewWidthSmallTabletMinus";
}
const stylesMode = {
    none: classNames("vkuiGroup__modeNone", 'vkuiInternalGroup--mode-none'),
    plain: classNames("vkuiGroup__modePlain", 'vkuiInternalGroup--mode-plain'),
    card: classNames("vkuiGroup__modeCard", 'vkuiInternalGroup--mode-card')
};
const stylesPadding = {
    s: "vkuiGroup__paddingS",
    m: "vkuiGroup__paddingM"
};
/**
 * Вычисляем mode для Group.
 */ function useGroupMode(forcedMode, viewWidth, legacySizeX, isInsideModal) {
    const { layout } = React.useContext(AppRootContext);
    if (forcedMode) {
        return forcedMode;
    }
    if (isInsideModal) {
        return 'plain';
    }
    if (layout) {
        return layout;
    }
    // TODO [>=10]: #9015 Удалить это условие
    if (legacySizeX !== undefined) {
        return legacySizeX === 'regular' ? 'card' : 'plain';
    }
    if (viewWidth !== 'none') {
        return viewWidth >= ViewWidth.SMALL_TABLET ? 'card' : 'plain';
    }
    return 'none';
}
const warn = warnOnce('Group');
export const GroupContainer = ({ children, separator = 'auto', mode: modeProps, padding = 'm', tabIndex: tabIndexProp, ...restProps })=>{
    const isInsideModal = useModalContext().id !== null;
    const { sizeX: legacySizeX, viewWidth = 'none' } = useAdaptivity();
    const mode = useGroupMode(modeProps, viewWidth, legacySizeX, isInsideModal);
    const isTabPanel = restProps.role === 'tabpanel';
    if (process.env.NODE_ENV === 'development' && isTabPanel && (!restProps['aria-controls'] || !restProps['id'])) {
        warn('При использовании роли "tabpanel" необходимо задать значение свойств "aria-controls" и "id"');
    }
    const tabIndex = isTabPanel && tabIndexProp === undefined ? 0 : tabIndexProp;
    let siblingSeparatorElement = null;
    switch(separator){
        case 'auto':
            siblingSeparatorElement = /*#__PURE__*/ _jsx("div", {
                className: "vkuiGroup__separatorSibling"
            });
            break;
        case 'show':
            siblingSeparatorElement = /*#__PURE__*/ _jsx("div", {
                className: classNames("vkuiGroup__separatorSibling", mode === 'plain' || mode === 'none' ? "vkuiGroup__separatorSiblingForced" : undefined)
            });
            break;
        case 'hide':
            break;
    }
    return /*#__PURE__*/ _jsxs(_Fragment, {
        children: [
            /*#__PURE__*/ _jsx(RootComponent, {
                Component: "section",
                ...restProps,
                tabIndex: tabIndex,
                baseClassName: classNames('vkuiInternalGroup', "vkuiGroup__host", getViewWidthClassName(viewWidth, legacySizeX), mode === 'plain' && isInsideModal && "vkuiGroup__modePlainInsideModal", stylesMode[mode], stylesPadding[padding]),
                children: children
            }),
            siblingSeparatorElement
        ]
    });
};
