// @vkontakte/vkui/dist/components/ScreenSpinner/ScreenSpinnerContainer.js
import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
import { classNames, hasReactNode } from "@vkontakte/vkjs";
import { RootComponent } from "../RootComponent/RootComponent.js";
import { Footnote } from "../Typography/Footnote/Footnote.js";
import { ScreenSpinnerContext } from "./context.js";
const stateClassNames = {
    cancelable: "vkuiScreenSpinner__stateCancelable",
    done: "vkuiScreenSpinner__stateDone",
    error: "vkuiScreenSpinner__stateError",
    custom: "vkuiScreenSpinner__stateCustom"
};
const modeClassNames = {
    shadow: "vkuiScreenSpinner__modeShadow",
    overlay: "vkuiScreenSpinner__modeOverlay"
};
export const ScreenSpinnerContainer = ({ state = 'loading', mode = 'shadow', customIcon, label, children, visibilityDelay, ...restProps })=>{
    return /*#__PURE__*/ _jsx(ScreenSpinnerContext.Provider, {
        value: {
            state,
            label,
            customIcon
        },
        children: /*#__PURE__*/ _jsxs(RootComponent, {
            baseClassName: classNames("vkuiScreenSpinner__host", modeClassNames[mode], state !== 'loading' && stateClassNames[state], hasReactNode(label) && "vkuiScreenSpinner__hasLabel", visibilityDelay && "vkuistyles__visibilityDelay"),
            ...restProps,
            children: [
                /*#__PURE__*/ _jsx("div", {
                    className: "vkuiScreenSpinner__iconSlot",
                    children: children
                }),
                hasReactNode(label) && /*#__PURE__*/ _jsx(Footnote, {
                    className: "vkuiScreenSpinner__label",
                    "aria-hidden": true,
                    children: label
                })
            ]
        })
    });
};
