// @vkontakte/vkui/dist/components/PopoutWrapper/PopoutWrapper.js
'use client';
import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
import { classNames } from "@vkontakte/vkjs";
import { RootComponent } from "../RootComponent/RootComponent.js";
const stylesAlignX = {
    center: "vkuiPopoutWrapper__alignXCenter",
    left: "vkuiPopoutWrapper__alignXLeft",
    right: "vkuiPopoutWrapper__alignXRight"
};
const stylesAlignY = {
    center: "vkuiPopoutWrapper__alignYCenter",
    top: "vkuiPopoutWrapper__alignYTop",
    bottom: "vkuiPopoutWrapper__alignYBottom"
};
const stylesStrategy = {
    fixed: "vkuiPopoutWrapper__fixed",
    absolute: "vkuiPopoutWrapper__absolute",
    none: undefined
};
/**
 * @see https://vkui.io/components/popout-wrapper
 */ export const PopoutWrapper = ({ alignY = 'center', alignX = 'center', closing = false, noBackground = false, strategy = 'fixed', children, onClick, zIndex = 'var(--vkui--z_index_popout)', ...restProps })=>/*#__PURE__*/ _jsx(RootComponent, {
        ...restProps,
        baseClassName: classNames("vkuiPopoutWrapper__host", stylesAlignY[alignY], stylesAlignX[alignX], closing ? "vkuiPopoutWrapper__closing" : "vkuiPopoutWrapper__opened", strategy && stylesStrategy[strategy], !noBackground && "vkuiPopoutWrapper__masked"),
        baseStyle: {
            zIndex
        },
        children: /*#__PURE__*/ _jsxs("div", {
            className: "vkuiPopoutWrapper__container",
            children: [
                /*#__PURE__*/ _jsx("div", {
                    className: "vkuiPopoutWrapper__overlay",
                    onClick: onClick
                }),
                /*#__PURE__*/ _jsx("div", {
                    className: "vkuiPopoutWrapper__content",
                    children: children
                })
            ]
        })
    });
