// @vkontakte/vkui/dist/components/RichCell/RichCell.js
'use client';
import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
import { classNames } from "@vkontakte/vkjs";
import { useAdaptivity } from "../../hooks/useAdaptivity.js";
import { warnOnce } from "../../lib/warnOnce.js";
import { Tappable } from "../Tappable/Tappable.js";
import { Subhead } from "../Typography/Subhead/Subhead.js";
import { RichCellIcon } from "./RichCellIcon/RichCellIcon.js";
const warn = warnOnce('RichCell');
const densityClassNames = {
    none: "vkuiRichCell__densityNone",
    compact: "vkuiRichCell__densityCompact"
};
const alignAfterClassNames = {
    start: "vkuiRichCell__alignAfterStart",
    center: "vkuiRichCell__alignAfterCenter",
    end: "vkuiRichCell__alignAfterEnd"
};
const alignBeforeClassNames = {
    start: "vkuiRichCell__alignBeforeStart",
    center: "vkuiRichCell__alignBeforeCenter",
    end: "vkuiRichCell__alignBeforeEnd"
};
const alignContentClassNames = {
    start: "vkuiRichCell__contentAlignStart",
    center: "vkuiRichCell__contentAlignCenter",
    end: "vkuiRichCell__contentAlignEnd"
};
/**
 * @see https://vkui.io/components/rich-cell
 */ export const RichCell = ({ overTitle, children, subtitle, extraSubtitle, before, after, afterCaption, bottom, actions, multiline, beforeAlign = 'start', contentAlign = 'start', afterAlign = 'start', meta, submeta, ...restProps })=>{
    const { density = 'none' } = useAdaptivity();
    /* istanbul ignore if: не проверяем в тестах */ if (process.env.NODE_ENV === 'development' && afterCaption) {
        warn('Свойство `afterCaption` устаревшее и будет удалено в `v10`, используйте `submeta`');
    }
    return /*#__PURE__*/ _jsxs(Tappable, {
        ...restProps,
        baseClassName: classNames("vkuiRichCell__host", !multiline && "vkuiRichCell__textEllipsis", density !== 'regular' && densityClassNames[density], (after || afterCaption) && alignAfterClassNames[afterAlign], before && alignBeforeClassNames[beforeAlign], alignContentClassNames[contentAlign]),
        children: [
            before && /*#__PURE__*/ _jsx("div", {
                className: "vkuiRichCell__before",
                children: before
            }),
            /*#__PURE__*/ _jsxs("div", {
                className: "vkuiRichCell__in",
                children: [
                    /*#__PURE__*/ _jsxs("div", {
                        className: "vkuiRichCell__content",
                        children: [
                            /*#__PURE__*/ _jsxs("div", {
                                className: "vkuiRichCell__contentBefore",
                                children: [
                                    overTitle && /*#__PURE__*/ _jsx(Subhead, {
                                        Component: "div",
                                        className: "vkuiRichCell__overTitle",
                                        children: overTitle
                                    }),
                                    /*#__PURE__*/ _jsx("div", {
                                        className: "vkuiRichCell__children",
                                        children: children
                                    }),
                                    subtitle && /*#__PURE__*/ _jsx("div", {
                                        className: "vkuiRichCell__subtitle",
                                        children: subtitle
                                    }),
                                    extraSubtitle && /*#__PURE__*/ _jsx(Subhead, {
                                        Component: "div",
                                        className: "vkuiRichCell__extraSubtitle",
                                        children: extraSubtitle
                                    })
                                ]
                            }),
                            (meta || submeta) && /*#__PURE__*/ _jsxs("div", {
                                className: "vkuiRichCell__contentMeta",
                                children: [
                                    meta && /*#__PURE__*/ _jsx("div", {
                                        className: "vkuiRichCell__afterChildren",
                                        children: meta
                                    }),
                                    submeta && /*#__PURE__*/ _jsx("div", {
                                        className: "vkuiRichCell__afterCaption",
                                        children: submeta
                                    })
                                ]
                            })
                        ]
                    }),
                    bottom && /*#__PURE__*/ _jsx("div", {
                        className: "vkuiRichCell__bottom",
                        children: bottom
                    }),
                    actions && /*#__PURE__*/ _jsx("div", {
                        className: "vkuiRichCell__actions",
                        children: actions
                    })
                ]
            }),
            (after || afterCaption) && /*#__PURE__*/ _jsxs("div", {
                className: "vkuiRichCell__contentAfter",
                children: [
                    after && /*#__PURE__*/ _jsx("div", {
                        className: "vkuiRichCell__afterChildren",
                        children: after
                    }),
                    afterCaption && /*#__PURE__*/ _jsx("div", {
                        className: "vkuiRichCell__afterCaption",
                        children: afterCaption
                    })
                ]
            })
        ]
    });
};
RichCell.Icon = RichCellIcon;
