// @vkontakte/vkui/dist/components/PanelHeaderContent/PanelHeaderContent.js
'use client';
import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
import { classNames, hasReactNode } from "@vkontakte/vkjs";
import { useAdaptivity } from "../../hooks/useAdaptivity.js";
import { usePlatform } from "../../hooks/usePlatform.js";
import { RootComponent } from "../RootComponent/RootComponent.js";
import { Tappable } from "../Tappable/Tappable.js";
import { Footnote } from "../Typography/Footnote/Footnote.js";
import { Text } from "../Typography/Text/Text.js";
const platformClassNames = {
    ios: "vkuiPanelHeaderContent__ios",
    android: "vkuiPanelHeaderContent__android",
    vkcom: "vkuiPanelHeaderContent__vkcom"
};
const densityClassNames = {
    none: "vkuiPanelHeaderContent__densityNone",
    compact: "vkuiPanelHeaderContent__densityCompact"
};
const PanelHeaderChildren = ({ hasSubtitle, hasBefore, children })=>{
    const platform = usePlatform();
    return hasSubtitle || hasBefore ? /*#__PURE__*/ _jsx(Text, {
        className: "vkuiPanelHeaderContent__childrenText",
        Component: "div",
        weight: platform === 'vkcom' ? '2' : undefined,
        children: children
    }) : /*#__PURE__*/ _jsx("div", {
        className: "vkuiPanelHeaderContent__childrenIn",
        children: children
    });
};
/**
 * @see https://vkui.io/components/panel-header#panel-header-content
 */ export const PanelHeaderContent = ({ aside, subtitle, before, children, onClick, ...restProps })=>{
    const { density = 'none' } = useAdaptivity();
    const InComponent = onClick ? Tappable : 'div';
    const rootProps = onClick ? {} : restProps;
    const platform = usePlatform();
    const inProps = onClick ? {
        ...restProps,
        onClick,
        activeEffectDelay: 200,
        hasActive: platform === 'ios',
        activeMode: 'opacity'
    } : {};
    return /*#__PURE__*/ _jsxs(RootComponent, {
        ...rootProps,
        baseClassName: classNames("vkuiPanelHeaderContent__host", platformClassNames.hasOwnProperty(platform) ? platformClassNames[platform] : platformClassNames.android, density !== 'regular' && densityClassNames[density]),
        children: [
            hasReactNode(before) && /*#__PURE__*/ _jsx("div", {
                className: "vkuiPanelHeaderContent__before",
                children: before
            }),
            /*#__PURE__*/ _jsxs(InComponent, {
                ...inProps,
                className: classNames("vkuiPanelHeaderContent__in", !before && platform !== 'android' && "vkuiPanelHeaderContent__inCentered"),
                children: [
                    hasReactNode(subtitle) && /*#__PURE__*/ _jsx(Footnote, {
                        className: "vkuiPanelHeaderContent__subtitle",
                        children: subtitle
                    }),
                    /*#__PURE__*/ _jsxs("div", {
                        className: "vkuiPanelHeaderContent__children",
                        children: [
                            /*#__PURE__*/ _jsx(PanelHeaderChildren, {
                                hasSubtitle: hasReactNode(subtitle),
                                hasBefore: hasReactNode(before),
                                children: children
                            }),
                            hasReactNode(aside) && /*#__PURE__*/ _jsx("div", {
                                className: "vkuiPanelHeaderContent__aside",
                                children: aside
                            })
                        ]
                    }),
                    hasReactNode(before) && /*#__PURE__*/ _jsx("div", {
                        className: "vkuiPanelHeaderContent__width"
                    })
                ]
            })
        ]
    });
};
