// @vkontakte/vkui/dist/components/SimpleCell/SimpleCell.js
'use client';
import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
import { classNames, hasReactNode } from "@vkontakte/vkjs";
import { useAdaptivity } from "../../hooks/useAdaptivity.js";
import { usePlatform } from "../../hooks/usePlatform.js";
import { Tappable } from "../Tappable/Tappable.js";
import { Footnote } from "../Typography/Footnote/Footnote.js";
import { Headline } from "../Typography/Headline/Headline.js";
import { Subhead } from "../Typography/Subhead/Subhead.js";
import { VisuallyHidden } from "../VisuallyHidden/VisuallyHidden.js";
import { Chevron } from "./Chevron/Chevron.js";
const densityClassNames = {
    none: "vkuiSimpleCell__densityNone",
    compact: "vkuiSimpleCell__densityCompact"
};
/**
 * @see https://vkui.io/components/simple-cell
 */ export const SimpleCell = ({ badgeBeforeTitle, badgeAfterTitle, badgeBeforeSubtitle, badgeAfterSubtitle, before, indicator, children, after, chevron, multiline, overTitle, subtitle, extraSubtitle, chevronSize = 'm', ...restProps })=>{
    const platform = usePlatform();
    const hasChevron = chevron === 'always' || chevron === 'auto' && platform === 'ios';
    const hasAfter = hasReactNode(after) || hasChevron;
    const { density = 'none' } = useAdaptivity();
    return /*#__PURE__*/ _jsxs(Tappable, {
        ...restProps,
        baseClassName: classNames("vkuiSimpleCell__host", restProps.disabled && "vkuiSimpleCell__disabled", density !== 'regular' && densityClassNames[density], multiline && "vkuiSimpleCell__mult"),
        children: [
            /*#__PURE__*/ _jsx("div", {
                className: classNames("vkuiSimpleCell__before", platform === 'ios' && "vkuiSimpleCell__beforeIos"),
                children: before
            }),
            /*#__PURE__*/ _jsxs("div", {
                className: "vkuiSimpleCell__middle",
                children: [
                    overTitle && /*#__PURE__*/ _jsxs(Subhead, {
                        Component: "span",
                        className: classNames("vkuiSimpleCell__text", "vkuiSimpleCell__overTitle"),
                        children: [
                            overTitle,
                            /*#__PURE__*/ _jsx(VisuallyHidden, {
                                children: " "
                            })
                        ]
                    }),
                    /*#__PURE__*/ _jsxs("div", {
                        className: "vkuiSimpleCell__content",
                        children: [
                            badgeBeforeTitle && /*#__PURE__*/ _jsx("span", {
                                className: "vkuiSimpleCell__badge",
                                children: badgeBeforeTitle
                            }),
                            /*#__PURE__*/ _jsxs(Headline, {
                                Component: "span",
                                className: "vkuiSimpleCell__children",
                                weight: "3",
                                children: [
                                    children,
                                    /*#__PURE__*/ _jsx(VisuallyHidden, {
                                        children: " "
                                    })
                                ]
                            }),
                            hasReactNode(badgeAfterTitle) && /*#__PURE__*/ _jsx("span", {
                                className: "vkuiSimpleCell__badge",
                                children: badgeAfterTitle
                            })
                        ]
                    }),
                    subtitle && /*#__PURE__*/ _jsxs("div", {
                        className: "vkuiSimpleCell__content",
                        children: [
                            badgeBeforeSubtitle && /*#__PURE__*/ _jsx("span", {
                                className: "vkuiSimpleCell__badge",
                                children: badgeBeforeSubtitle
                            }),
                            /*#__PURE__*/ _jsxs(Footnote, {
                                normalize: false,
                                className: classNames("vkuiSimpleCell__text", "vkuiSimpleCell__subtitle"),
                                children: [
                                    subtitle,
                                    /*#__PURE__*/ _jsx(VisuallyHidden, {
                                        children: " "
                                    })
                                ]
                            }),
                            badgeAfterSubtitle && /*#__PURE__*/ _jsx("span", {
                                className: "vkuiSimpleCell__badge",
                                children: badgeAfterSubtitle
                            })
                        ]
                    }),
                    extraSubtitle && /*#__PURE__*/ _jsxs(Footnote, {
                        className: classNames("vkuiSimpleCell__text", "vkuiSimpleCell__extraSubtitle"),
                        children: [
                            extraSubtitle,
                            /*#__PURE__*/ _jsx(VisuallyHidden, {
                                children: " "
                            })
                        ]
                    })
                ]
            }),
            hasReactNode(indicator) && /*#__PURE__*/ _jsx(Headline, {
                Component: "span",
                weight: "3",
                className: "vkuiSimpleCell__indicator",
                children: indicator
            }),
            hasAfter && /*#__PURE__*/ _jsxs("div", {
                className: classNames("vkuiSimpleCell__after", 'vkuiInternalSimpleCell__after'),
                children: [
                    after,
                    hasChevron && /*#__PURE__*/ _jsx(Chevron, {
                        size: chevronSize,
                        className: "vkuiSimpleCell__chevronIcon"
                    })
                ]
            })
        ]
    });
};
