// @vkontakte/vkui/dist/components/AppRoot/AppRootStyleContainer/AppRootStyleContainer.js
'use client';
import { jsx as _jsx } from "react/jsx-runtime";
import * as React from "react";
import { classNames } from "@vkontakte/vkjs";
import { useAdaptivity } from "../../../hooks/useAdaptivity.js";
import { useTokensClassName } from "../../../lib/tokens/useTokenClassName.js";
import { useConfigProvider } from "../../ConfigProvider/ConfigProviderContext.js";
import { RootComponent } from "../../RootComponent/RootComponent.js";
import { AppRootContext } from "../AppRootContext.js";
import { getSafeAreaInsetsAsCssVariables, getUserSelectModeClassName } from "../helpers.js";
const densityClassNames = {
    none: "vkuiAppRootStyleContainer__densityNone",
    compact: "vkuiAppRootStyleContainer__densityCompact"
};
export function useAppRootStyles() {
    const { safeAreaInsets, mode, userSelectMode } = React.useContext(AppRootContext);
    const { density = 'none', hasPointer } = useAdaptivity();
    const { isWebView } = useConfigProvider();
    const userSelectModeClassName = getUserSelectModeClassName({
        userSelectMode,
        isWebView,
        hasPointer
    });
    const tokensClassName = useTokensClassName();
    return {
        style: safeAreaInsets ? getSafeAreaInsetsAsCssVariables(safeAreaInsets) : undefined,
        className: classNames("vkuiAppRootStyleContainer__host", mode === 'embedded' && "vkuiAppRootStyleContainer__embedded", density !== 'regular' && densityClassNames[density], userSelectModeClassName, tokensClassName)
    };
}
/**
 * Специальный контейнер для переиспользования стилей, токенов и safe-area-inset в:
 * - точке монтирования приложения – `AppRoot`;
 * - точке монтирования порталов для модальных окон – `AppRootPortal`.
 *
 * @private
 */ export function AppRootStyleContainer(props) {
    const { style: appRootStyle, className: appRootClassName } = useAppRootStyles();
    return /*#__PURE__*/ _jsx(RootComponent, {
        baseClassName: appRootClassName,
        baseStyle: appRootStyle,
        ...props
    });
}
