// @vkontakte/vkui/dist/cssm/hooks/useAdaptivityConditionalRender/constants.js
import styles from "../../styles/adaptivity.module.css";
export const forcedProps = {
    className: ''
};
/**
 * @deprecated Since 8.0.0. см. https://github.com/VKCOM/VKUI/issues/9015
 * TODO [>=10]: #9015 Удалить константу.
 */ export const sizeXCompactMediaQueryProps = {
    className: styles.viewWidthSmallTabletMinusMq
};
/**
 * @deprecated Since 8.0.0. см. https://github.com/VKCOM/VKUI/issues/9015
 * TODO [>=10]: #9015 Удалить константу.
 */ export const sizeXRegularMediaQueryProps = {
    className: styles.viewWidthSmallTabletPlusMq
};
export const densityCompactMediaQueryProps = {
    className: styles.densityCompactMq
};
export const densityRegularMediaQueryProps = {
    className: styles.densityRegularMq
};
export const viewWidthMediaQueryMapProps = {
    smallTabletMinus: {
        className: styles.viewWidthSmallTabletMinusMq
    },
    smallTabletPlus: {
        className: styles.viewWidthSmallTabletPlusMq
    },
    tabletMinus: {
        className: styles.viewWidthTabletMinusMq
    },
    tabletPlus: {
        className: styles.viewWidthTabletPlusMq
    }
};
export const deviceTypeMediaQueryMapProps = {
    mobile: {
        className: styles.deviceTypeMobileMq
    },
    desktop: {
        className: styles.deviceTypeDesktopMq
    }
};
