// @vkontakte/vkui/dist/components/Panel/Panel.js
'use client';
import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
import * as React from "react";
import { classNames } from "@vkontakte/vkjs";
import { useAdaptivity } from "../../hooks/useAdaptivity.js";
import { ViewWidth } from "../../lib/adaptivity/index.js";
import { AppRootContext } from "../AppRoot/AppRootContext.js";
import { NavPanelIdContext } from "../NavIdContext/NavIdContext.js";
import { OnboardingTooltipContainer } from "../OnboardingTooltip/OnboardingTooltipContainer.js";
import { RootComponent } from "../RootComponent/RootComponent.js";
import { Touch } from "../Touch/Touch.js";
function getViewWidthClassName(viewWidth, legacySizeX) {
    // TODO [>=10]: #9015 Удалить это условие
    if (legacySizeX !== undefined) {
        return legacySizeX === 'regular' ? "vkuiPanel__viewWidthSmallTabletPlus" : "vkuiPanel__viewWidthSmallTabletMinus";
    }
    if (viewWidth === 'none') {
        return classNames("vkuiPanel__viewWidthNone", 'vkuiInternalGroup--viewWidth-none');
    }
    return viewWidth >= ViewWidth.SMALL_TABLET ? "vkuiPanel__viewWidthSmallTabletPlus" : "vkuiPanel__viewWidthSmallTabletMinus";
}
const stylesMode = {
    none: "vkuiPanel__modeNone",
    plain: "vkuiPanel__modePlain",
    card: "vkuiPanel__modeCard"
};
/**
 * @see https://vkui.io/components/panel
 */ export const Panel = ({ centered = false, children, nav, mode: modeProp, disableBackground, ...restProps })=>{
    const { sizeX: legacySizeX, viewWidth = 'none' } = useAdaptivity();
    const mode = usePanelMode(modeProp, viewWidth, legacySizeX);
    return /*#__PURE__*/ _jsx(NavPanelIdContext.Provider, {
        value: restProps.id || nav,
        children: /*#__PURE__*/ _jsx(RootComponent, {
            ...restProps,
            baseClassName: classNames("vkuiPanel__host", getViewWidthClassName(viewWidth, legacySizeX), centered && 'vkuiInternalPanel--centered', disableBackground && "vkuiPanel__disableBackground", stylesMode[mode]),
            children: /*#__PURE__*/ _jsxs(Touch, {
                Component: OnboardingTooltipContainer,
                className: classNames("vkuiPanel__in", 'vkuiInternalPanel__in'),
                children: [
                    /*#__PURE__*/ _jsx("div", {
                        className: "vkuiPanel__inBefore"
                    }),
                    centered ? /*#__PURE__*/ _jsx("div", {
                        className: "vkuiPanel__centered",
                        children: children
                    }) : children,
                    /*#__PURE__*/ _jsx("div", {
                        className: "vkuiPanel__inAfter"
                    })
                ]
            })
        })
    });
};
function usePanelMode(modeProp, viewWidth, legacySizeX) {
    const { layout } = React.useContext(AppRootContext);
    if (modeProp) {
        return modeProp;
    }
    if (layout) {
        return layout;
    }
    // TODO [>=10]: #9015 Удалить это условие
    if (legacySizeX !== undefined) {
        return legacySizeX === 'regular' ? 'card' : 'plain';
    }
    if (viewWidth !== 'none') {
        return viewWidth >= ViewWidth.SMALL_TABLET ? 'card' : 'plain';
    }
    return 'none';
}
