// @vkontakte/vkui/dist/components/Link/Link.js
import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
import { classNames } from "@vkontakte/vkjs";
import { Tappable } from "../Tappable/Tappable.js";
/**
 * @see https://vkui.io/components/link
 */ export const Link = ({ before: beforeProp, after: afterProp, noUnderline, hasVisited, children, ...restProps })=>{
    const before = beforeProp ? /*#__PURE__*/ _jsx("span", {
        className: "vkuiLink__before",
        children: beforeProp
    }) : null;
    const after = afterProp ? /*#__PURE__*/ _jsx("span", {
        className: "vkuiLink__after",
        children: afterProp
    }) : null;
    return /*#__PURE__*/ _jsxs(Tappable, {
        activeMode: "opacity",
        hoverMode: "vkuiLink__hover",
        focusVisibleMode: "outside",
        DefaultComponent: "span",
        ...restProps,
        baseClassName: classNames("vkuiLink__host", hasVisited && "vkuiLink__hasVisited", !noUnderline && "vkuiLink__withUnderline"),
        children: [
            before,
            children,
            after
        ]
    });
};
