// @vkontakte/vkui/dist/components/Checkbox/Checkbox.js
'use client';
import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
import { hasReactNode } from "@vkontakte/vkjs";
import { useMergeProps } from "../../hooks/useMergeProps.js";
import { warnOnce } from "../../lib/warnOnce.js";
import { withLabelClickWrapper } from "../../lib/withLabelClickWrapper.js";
import { SelectionControl } from "../SelectionControl/SelectionControl.js";
import { SelectionControlLabel } from "../SelectionControl/SelectionControlLabel/SelectionControlLabel.js";
import { CheckboxInput } from "./CheckboxInput/CheckboxInput.js";
import { CheckboxSimple } from "./CheckboxSimple/CheckboxSimple.js";
const warn = warnOnce('Checkbox');
const CheckboxComponent = ({ // CheckboxProps
getRef, description, titleAfter, noPadding, children, // CheckboxInputProps
indeterminate, defaultIndeterminate, IconOnCompact, IconOnRegular, IconOffCompact, IconOffRegular, IconIndeterminate, // Tappable props
hoverMode, activeMode, hasHover, hasActive, focusVisibleMode, // Input props
checked, defaultChecked, disabled, readOnly, required, autoFocus, id, name, value, form, onChange, onFocus, onBlur, slotProps, ...restProps })=>{
    const { onClick, ...rootRest } = useMergeProps(restProps, slotProps === null || slotProps === void 0 ? void 0 : slotProps.root);
    const inputRest = useMergeProps({
        getRootRef: getRef,
        checked,
        defaultChecked,
        disabled,
        readOnly,
        required,
        autoFocus,
        id,
        name,
        value,
        form,
        onChange,
        onFocus,
        onBlur
    }, slotProps === null || slotProps === void 0 ? void 0 : slotProps.input);
    return /*#__PURE__*/ _jsxs(SelectionControl, {
        disabled: inputRest.disabled,
        hoverMode: hoverMode,
        activeMode: activeMode,
        hasHover: hasHover,
        hasActive: hasActive,
        focusVisibleMode: focusVisibleMode,
        noPadding: noPadding,
        onClick: withLabelClickWrapper(onClick),
        ...rootRest,
        children: [
            /*#__PURE__*/ _jsx(CheckboxInput, {
                indeterminate: indeterminate,
                defaultIndeterminate: defaultIndeterminate,
                IconIndeterminate: IconIndeterminate,
                IconOnCompact: IconOnCompact,
                IconOnRegular: IconOnRegular,
                IconOffCompact: IconOffCompact,
                IconOffRegular: IconOffRegular,
                slotProps: {
                    input: inputRest
                }
            }),
            /*#__PURE__*/ _jsx(SelectionControlLabel, {
                titleAfter: titleAfter,
                description: description,
                children: children
            })
        ]
    });
};
/**
 * @see https://vkui.io/components/checkbox
 */ export const Checkbox = (props)=>{
    if (process.env.NODE_ENV === 'development' && props.getRef) {
        warn('Свойство `getRef` устаревшее, используйте `slotProps={ input: { getRootRef: ... } }`');
    }
    const simple = !(hasReactNode(props.children) || hasReactNode(props.description));
    if (simple) {
        return /*#__PURE__*/ _jsx(CheckboxSimple, {
            ...props
        });
    }
    return /*#__PURE__*/ _jsx(CheckboxComponent, {
        ...props
    });
};
Checkbox.Input = CheckboxInput;
