// @vkontakte/vkui/dist/components/Button/Button.js
'use client';
import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
import * as React from "react";
import { classNames, hasReactNode } from "@vkontakte/vkjs";
import { useAdaptivity } from "../../hooks/useAdaptivity.js";
import { usePlatform } from "../../hooks/usePlatform.js";
import { Spinner } from "../Spinner/Spinner.js";
import { Tappable } from "../Tappable/Tappable.js";
const stylesElevation = {
    '1': "vkuiButton__elevation1",
    '2': "vkuiButton__elevation2",
    '3': "vkuiButton__elevation3",
    '4': "vkuiButton__elevation4"
};
const stylesSize = {
    s: "vkuiButton__sizeS",
    m: "vkuiButton__sizeM",
    l: "vkuiButton__sizeL"
};
const stylesMode = {
    primary: "vkuiButton__modePrimary",
    secondary: "vkuiButton__modeSecondary",
    tertiary: "vkuiButton__modeTertiary",
    outline: "vkuiButton__modeOutline",
    link: "vkuiButton__modeLink"
};
const stylesAppearance = {
    'accent': "vkuiButton__appearanceAccent",
    'positive': "vkuiButton__appearancePositive",
    'negative': "vkuiButton__appearanceNegative",
    'neutral': "vkuiButton__appearanceNeutral",
    'overlay': "vkuiButton__appearanceOverlay",
    'accent-invariable': "vkuiButton__appearanceAccentInvariable"
};
const stylesAlign = {
    left: "vkuiButton__alignLeft",
    right: "vkuiButton__alignRight"
};
const densityClassNames = {
    none: "vkuiButton__densityNone",
    regular: "vkuiButton__densityRegular"
};
/**
 * @see https://vkui.io/components/button
 */ export const Button = ({ size = 's', mode = 'primary', appearance = 'accent', stretched = false, align = 'center', children, before, after, getRootRef, loading, loadingLabel = 'Загрузка...', onClick, disableSpinnerAnimation, rounded, disabled, href, 'aria-label': ariaLabelProp, elevation, ...restProps })=>{
    const hasIconOnly = !children && Boolean(after) !== Boolean(before);
    const { density = 'none' } = useAdaptivity();
    const platform = usePlatform();
    const isDisabled = disabled || loading;
    const hasHref = href !== undefined;
    const ariaLabel = loading ? loadingLabel : ariaLabelProp;
    const buttonProps = React.useMemo(()=>{
        if (hasHref) {
            return isDisabled ? {
                // Для disabled/loading ссылок нужно удалить href и добавить role="link"
                // согласно https://w3c.github.io/html-aria/#example-communicate-a-disabled-link-with-aria
                'role': 'link',
                'Component': 'a',
                'aria-disabled': isDisabled
            } : {
                href,
                'aria-disabled': isDisabled
            };
        } else {
            return {
                Component: 'button',
                disabled
            };
        }
    }, [
        disabled,
        hasHref,
        href,
        isDisabled
    ]);
    return /*#__PURE__*/ _jsxs(Tappable, {
        hoverMode: "vkuiButton__hover",
        activeMode: "vkuiButton__active",
        focusVisibleMode: "outside",
        "aria-busy": loading,
        hasHover: !loading,
        hasActive: !loading,
        ...buttonProps,
        ...restProps,
        "aria-label": ariaLabel,
        onClick: isDisabled ? undefined : onClick,
        baseClassName: classNames("vkuiButton__host", stylesSize[size], stylesMode[mode], stylesAppearance[appearance], align !== 'center' && stylesAlign[align], density !== 'compact' && densityClassNames[density], platform === 'ios' && "vkuiButton__ios", stretched && "vkuiButton__stretched", hasIconOnly && !stretched && "vkuiButton__singleIcon", loading && "vkuiButton__loading", rounded && "vkuiButton__rounded", disabled && "vkuiButton__disabled", elevation && stylesElevation[elevation]),
        getRootRef: getRootRef,
        children: [
            loading && /*#__PURE__*/ _jsx(Spinner, {
                size: "s",
                className: "vkuiButton__spinner",
                disableAnimation: disableSpinnerAnimation,
                noColor: true,
                "aria-hidden": "true"
            }),
            /*#__PURE__*/ _jsxs("span", {
                className: "vkuiButton__in",
                children: [
                    hasReactNode(before) && /*#__PURE__*/ _jsx("span", {
                        className: "vkuiButton__before",
                        role: "presentation",
                        "data-testid": process.env.NODE_ENV === 'test' ? 'before' : undefined,
                        children: before
                    }),
                    hasReactNode(children) && /*#__PURE__*/ _jsx("span", {
                        className: "vkuiButton__content",
                        "data-testid": process.env.NODE_ENV === 'test' ? 'children' : undefined,
                        children: children
                    }),
                    hasReactNode(after) && /*#__PURE__*/ _jsx("span", {
                        className: "vkuiButton__after",
                        role: "presentation",
                        "data-testid": process.env.NODE_ENV === 'test' ? 'after' : undefined,
                        children: after
                    })
                ]
            })
        ]
    });
};
