// @vkontakte/vkui/dist/components/ModalRoot/ModalRoot.js
'use client';
import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
import * as React from "react";
import { AppRootPortal } from "../AppRoot/AppRootPortal.js";
import { ModalOverlay } from "../ModalOverlay/ModalOverlay.js";
import { ModalRootContext, ModalRootOverlayContext } from "./ModalRootContext.js";
/**
 * @deprecated Since 8.0.0. Компонент устарел и будет удален в **VKUI v10**.
 * Вместо него вы можете использовать хук [useModalManager](/components/use-modal-manager).
 * @see https://vkui.io/components/modal-root
 */ export const ModalRoot = ({ activeModal, children, modalOverlayTestId, noFocusToDialog, disableModalOverlay, usePortal, onOpen, onOpened, onClose, onClosed, onOverlayClosed, onOverlayShowed, disableCloseAnimation, disableOpenAnimation })=>{
    const contextValue = React.useMemo(()=>({
            isInsideModal: true,
            // base props
            activeModal,
            modalOverlayTestId,
            noFocusToDialog,
            disableModalOverlay,
            disableCloseAnimation,
            disableOpenAnimation,
            // callbacks
            onOpen,
            onOpened,
            onClose,
            onClosed
        }), [
        activeModal,
        disableCloseAnimation,
        disableModalOverlay,
        disableOpenAnimation,
        modalOverlayTestId,
        noFocusToDialog,
        onClose,
        onClosed,
        onOpen,
        onOpened
    ]);
    const modalOverlayRef = React.useRef(null);
    return /*#__PURE__*/ _jsx(AppRootPortal, {
        usePortal: usePortal,
        children: /*#__PURE__*/ _jsx(ModalRootContext.Provider, {
            value: contextValue,
            children: /*#__PURE__*/ _jsxs(ModalRootOverlayContext.Provider, {
                value: modalOverlayRef,
                children: [
                    !disableModalOverlay && /*#__PURE__*/ _jsx(ModalOverlay, {
                        position: "fixed",
                        visible: typeof activeModal === 'string',
                        getRootRef: modalOverlayRef,
                        onShowed: onOverlayShowed,
                        onClosed: onOverlayClosed,
                        disableCloseAnimation: disableCloseAnimation,
                        disableOpenAnimation: disableOpenAnimation
                    }),
                    children
                ]
            })
        })
    });
};
