// @vkontakte/vkui/dist/components/SegmentedControl/SegmentedControlOption/SegmentedControlOption.js
'use client';
import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
import { hasReactNode } from "@vkontakte/vkjs";
import { Clickable } from "../../Clickable/Clickable.js";
import { Headline } from "../../Typography/Headline/Headline.js";
import { VisuallyHidden } from "../../VisuallyHidden/VisuallyHidden.js";
export const SegmentedControlOption = ({ getRef, children, getRootRef, before, rootProps, inputProps })=>/*#__PURE__*/ _jsxs(Clickable, {
        Component: "label",
        baseClassName: "vkuiSegmentedControlOption__host",
        hoverClassName: "vkuiSegmentedControlOption__hover",
        activeClassName: "vkuiSegmentedControlOption__hover",
        getRootRef: getRootRef,
        ...rootProps,
        children: [
            inputProps && /*#__PURE__*/ _jsx(VisuallyHidden, {
                ...inputProps,
                Component: "input",
                getRootRef: getRef,
                type: "radio"
            }),
            hasReactNode(before) && /*#__PURE__*/ _jsx("div", {
                className: "vkuiSegmentedControlOption__before",
                children: before
            }),
            /*#__PURE__*/ _jsx(Headline, {
                level: "2",
                weight: "2",
                children: children
            })
        ]
    });
