// @vkontakte/vkui/dist/hooks/useSnackbarManager/helpers/snackbarStateHelpers.js
export const addSnackbarToState = (state, snackbar)=>{
    return {
        ...state,
        snackbars: [
            ...state.snackbars,
            snackbar
        ]
    };
};
export const removeSnackbarFromState = (state, id)=>{
    return {
        snackbars: state.snackbars.filter((snackbar)=>snackbar.id !== id),
        snackbarsToClose: (()=>{
            const newSet = new Set(state.snackbarsToClose);
            newSet.delete(id);
            return newSet;
        })()
    };
};
export const updateSnackbarPropsInState = (state, id, config)=>{
    return {
        ...state,
        snackbars: state.snackbars.map((snackbar)=>snackbar.id === id ? {
                ...snackbar,
                snackbarProps: {
                    ...snackbar.snackbarProps,
                    ...config
                }
            } : snackbar)
    };
};
export const closeSnackbarInState = (state, id)=>{
    const newSnackbarsToClose = new Set(state.snackbarsToClose);
    newSnackbarsToClose.add(id);
    return {
        ...state,
        snackbarsToClose: newSnackbarsToClose
    };
};
export const closeAllSnackbarsInState = (state, showedSnackbars)=>{
    return {
        snackbars: state.snackbars.filter(({ id })=>showedSnackbars.has(id)),
        snackbarsToClose: new Set(showedSnackbars)
    };
};
export const closeOverflowedSnackbarsInState = (state, placementSnackbars)=>{
    const snackbarToClose = placementSnackbars.find((snackbar)=>{
        return !state.snackbarsToClose.has(snackbar.id);
    });
    if (snackbarToClose) {
        return closeSnackbarInState(state, snackbarToClose.id);
    }
    return state;
};
