// @vkontakte/vkui/dist/components/Typography/DisplayTitle/DisplayTitle.js
'use client';
import { jsx as _jsx } from "react/jsx-runtime";
import { classNames } from "@vkontakte/vkjs";
import { useAdaptivity } from "../../../hooks/useAdaptivity.js";
import { Typography } from "../Typography.js";
const stylesLevel = {
    '1': "vkuiDisplayTitle__level1",
    '2': "vkuiDisplayTitle__level2",
    '3': "vkuiDisplayTitle__level3",
    '4': "vkuiDisplayTitle__level4"
};
const densityClassNames = {
    none: "vkuiDisplayTitle__densityNone",
    compact: "vkuiDisplayTitle__densityCompact"
};
/**
 * Используется для крупных заголовков.
 *
 * @see https://vkui.io/components/typography#display-title
 */ export const DisplayTitle = ({ className, level = '1', Component = 'span', normalize = true, inline = false, ...restProps })=>{
    const { density = 'none' } = useAdaptivity();
    return /*#__PURE__*/ _jsx(Typography, {
        Component: Component,
        normalize: normalize,
        inline: inline,
        className: classNames(className, density !== 'regular' && densityClassNames[density], stylesLevel[level]),
        ...restProps
    });
};
