// @vkontakte/vkui/dist/components/Input/Input.js
'use client';
import { jsx as _jsx } from "react/jsx-runtime";
import { classNames } from "@vkontakte/vkjs";
import { useAdaptivity } from "../../hooks/useAdaptivity.js";
import { useMergeProps } from "../../hooks/useMergeProps.js";
import { warnOnce } from "../../lib/warnOnce.js";
import { FormField } from "../FormField/FormField.js";
import { UnstyledTextField } from "../UnstyledTextField/UnstyledTextField.js";
const warn = warnOnce('Input');
const densityClassNames = {
    none: "vkuiInput__densityNone",
    compact: "vkuiInput__densityCompact"
};
/**
 * @see https://vkui.io/components/input
 */ export const Input = ({ // InputProps
align = 'left', getRef, // FormFieldProps
before, after, status, mode, // input props
autoComplete, autoCapitalize, autoCorrect, disabled, list, max, maxLength, min, minLength, multiple, name, pattern, placeholder, readOnly, required, step, type = 'text', value, onChange, onFocus, onBlur, form, id, inputMode, defaultValue, autoFocus, tabIndex, spellCheck, slotProps, ...restProps })=>{
    /* istanbul ignore if: не проверяем в тестах */ if (process.env.NODE_ENV === 'development' && getRef) {
        warn('Свойство `getRef` устаревшее, используйте `slotProps={ input: { getRootRef: ... } }`');
    }
    const { density = 'none' } = useAdaptivity();
    const { className, ...rootProps } = useMergeProps(restProps, slotProps === null || slotProps === void 0 ? void 0 : slotProps.root);
    const inputRest = useMergeProps({
        className: "vkuiInput__el",
        getRootRef: getRef,
        autoComplete,
        autoCapitalize,
        autoCorrect,
        disabled,
        list,
        max,
        maxLength,
        min,
        minLength,
        multiple,
        name,
        pattern,
        placeholder,
        readOnly,
        required,
        step,
        type,
        value,
        onChange,
        onFocus,
        onBlur,
        form,
        id,
        inputMode,
        defaultValue,
        autoFocus,
        tabIndex,
        spellCheck
    }, slotProps === null || slotProps === void 0 ? void 0 : slotProps.input);
    return /*#__PURE__*/ _jsx(FormField, {
        className: classNames("vkuiInput__host", align === 'right' && "vkuiInput__alignRight", align === 'center' && "vkuiInput__alignCenter", density !== 'regular' && densityClassNames[density], before && "vkuiInput__hasBefore", after && "vkuiInput__hasAfter", className),
        before: before,
        after: after,
        disabled: inputRest.disabled,
        mode: mode,
        status: status,
        ...rootProps,
        children: /*#__PURE__*/ _jsx(UnstyledTextField, {
            as: "input",
            ...inputRest
        })
    });
};
