// @vkontakte/vkui/dist/hooks/useAdaptivityConditionalRender/constants.js
export const forcedProps = {
    className: ''
};
/**
 * @deprecated Since 8.0.0. см. https://github.com/VKCOM/VKUI/issues/9015
 * TODO [>=10]: #9015 Удалить константу.
 */ export const sizeXCompactMediaQueryProps = {
    className: "vkuistyles__viewWidthSmallTabletMinusMq"
};
/**
 * @deprecated Since 8.0.0. см. https://github.com/VKCOM/VKUI/issues/9015
 * TODO [>=10]: #9015 Удалить константу.
 */ export const sizeXRegularMediaQueryProps = {
    className: "vkuistyles__viewWidthSmallTabletPlusMq"
};
export const densityCompactMediaQueryProps = {
    className: "vkuistyles__densityCompactMq"
};
export const densityRegularMediaQueryProps = {
    className: "vkuistyles__densityRegularMq"
};
export const viewWidthMediaQueryMapProps = {
    smallTabletMinus: {
        className: "vkuistyles__viewWidthSmallTabletMinusMq"
    },
    smallTabletPlus: {
        className: "vkuistyles__viewWidthSmallTabletPlusMq"
    },
    tabletMinus: {
        className: "vkuistyles__viewWidthTabletMinusMq"
    },
    tabletPlus: {
        className: "vkuistyles__viewWidthTabletPlusMq"
    }
};
export const deviceTypeMediaQueryMapProps = {
    mobile: {
        className: "vkuistyles__deviceTypeMobileMq"
    },
    desktop: {
        className: "vkuistyles__deviceTypeDesktopMq"
    }
};
