// @vkontakte/vkui/dist/components/FormField/FormField.js
'use client';
import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
import * as React from "react";
import { classNames } from "@vkontakte/vkjs";
import { useAdaptivity } from "../../hooks/useAdaptivity.js";
import { useExternRef } from "../../hooks/useExternRef.js";
import { useFocusVisibleClassName } from "../../hooks/useFocusVisibleClassName.js";
import { useFocusWithin } from "../../hooks/useFocusWithin.js";
const densityClassNames = {
    none: "vkuiFormField__densityNone",
    compact: "vkuiFormField__densityCompact"
};
const stylesStatus = {
    error: "vkuiFormField__statusError",
    valid: "vkuiFormField__statusValid"
};
const iconAlignClassNames = {
    center: undefined,
    start: "vkuiFormField__iconAlignStart",
    end: "vkuiFormField__iconAlignEnd"
};
const renderIcon = (icon, align, className)=>{
    return /*#__PURE__*/ _jsx("div", {
        className: "vkuiFormField__iconWrapper",
        children: /*#__PURE__*/ _jsx("span", {
            className: classNames(iconAlignClassNames[align], className),
            children: icon
        })
    });
};
/**
 * @see https://vkui.io/components/form-field
 */ export const FormField = ({ Component = 'span', status = 'default', children, getRootRef, before, after, beforeAlign = 'center', afterAlign = 'center', disabled, mode = 'default', className, maxHeight, style, onMouseEnter, onMouseLeave, ...restProps })=>{
    const elRef = useExternRef(getRootRef);
    const { density = 'none' } = useAdaptivity();
    const [hover, setHover] = React.useState(false);
    const focusWithin = useFocusWithin(elRef);
    const focusVisibleClassNames = useFocusVisibleClassName({
        focusVisible: focusWithin,
        mode: "vkuiFormField__focusVisible"
    });
    const handleMouseEnter = (e)=>{
        e.stopPropagation();
        setHover(true);
        onMouseEnter === null || onMouseEnter === void 0 ? void 0 : onMouseEnter(e);
    };
    const handleMouseLeave = (e)=>{
        e.stopPropagation();
        setHover(false);
        onMouseLeave === null || onMouseLeave === void 0 ? void 0 : onMouseLeave(e);
    };
    return /*#__PURE__*/ _jsxs(Component, {
        ...restProps,
        ref: elRef,
        style: {
            maxHeight,
            ...style
        },
        onMouseEnter: handleMouseEnter,
        onMouseLeave: handleMouseLeave,
        className: classNames("vkuiFormField__host", mode === 'default' && "vkuiFormField__modeDefault", status !== 'default' && stylesStatus[status], density !== 'regular' && densityClassNames[density], disabled && "vkuiFormField__disabled", !disabled && hover && "vkuiFormField__hover", focusVisibleClassNames, className),
        children: [
            /*#__PURE__*/ _jsxs("div", {
                className: "vkuiFormField__scrollContainer",
                children: [
                    before && renderIcon(before, beforeAlign, "vkuiFormField__before"),
                    /*#__PURE__*/ _jsx("div", {
                        className: "vkuiFormField__content",
                        children: children
                    }),
                    after && renderIcon(after, afterAlign, classNames("vkuiFormField__after", 'vkuiInternalFormField__after'))
                ]
            }),
            /*#__PURE__*/ _jsx("span", {
                "aria-hidden": true,
                className: "vkuiFormField__border"
            })
        ]
    });
};
