// @vkontakte/vkui/dist/cssm/hooks/useSnackbarManager/components/SnackbarAnimatedWrapper.js
'use client';
import { jsx as _jsx } from "react/jsx-runtime";
import { useEffect, useRef } from "react";
import { classNames } from "@vkontakte/vkjs";
import { useCSSKeyframesAnimationController } from "../../../lib/animation/index.js";
import { usePlatform } from "../../usePlatform.js";
import styles from "./SnackbarAnimatedWrapper.module.css";
const animationStateClassNames = {
    enter: styles.stateEnter,
    entering: styles.stateEnter,
    entered: styles.stateEnter,
    exit: styles.stateExit,
    exiting: styles.stateExit,
    exited: undefined
};
const placementClassNames = {
    top: styles.placementTop,
    bottom: styles.placementBottom
};
export const SnackbarAnimatedWrapper = ({ snackbarItem, close = false, onClosed, placement })=>{
    const ref = useRef(null);
    const platform = usePlatform();
    const [animationState, animationHandlers] = useCSSKeyframesAnimationController(close ? 'exit' : 'enter', {
        onExited: ()=>onClosed(snackbarItem.id)
    });
    useEffect(()=>{
        if (ref.current && animationState === 'entered') {
            ref.current.style.setProperty('--vkui_internal--SnackbarAnimatedWrapper_height', `${ref.current.offsetHeight}px`);
        }
    }, [
        animationState
    ]);
    if (animationState === 'exited') {
        return null;
    }
    return /*#__PURE__*/ _jsx("div", {
        ref: ref,
        className: classNames(styles.host, animationStateClassNames[animationState], platform === 'ios' && styles.ios, placementClassNames[placement]),
        ...animationHandlers,
        children: /*#__PURE__*/ _jsx(snackbarItem.component, {
            ...snackbarItem.additionalProps,
            id: snackbarItem.id,
            snackbarProps: snackbarItem.snackbarProps,
            close: snackbarItem.close,
            update: snackbarItem.update
        })
    });
};
