// @vkontakte/vkui/dist/components/File/File.js
'use client';
import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
import { useMergeProps } from "../../hooks/useMergeProps.js";
import { warnOnce } from "../../lib/warnOnce.js";
import { withLabelClickWrapper } from "../../lib/withLabelClickWrapper.js";
import { Button } from "../Button/Button.js";
import { VisuallyHidden } from "../VisuallyHidden/VisuallyHidden.js";
const warn = warnOnce('File');
/**
 * @see https://vkui.io/components/file
 */ export const File = ({ // FileProps
getRootRef, children = 'Выберите файл', getRef, // VKUIButtonProps
align = 'left', size, mode, stretched, before, after, loading, appearance, // Input props
disabled, readOnly, required, autoFocus, id, name, value, accept, capture, multiple, form, onChange, onFocus, onBlur, slotProps, ...restProps })=>{
    /* istanbul ignore if: не проверяем в тестах */ if (process.env.NODE_ENV === 'development' && getRef) {
        warn('Свойство `getRef` устаревшее, используйте `slotProps={ input: { getRootRef: ... } }`');
    }
    const { onClick, ...rootRest } = useMergeProps({
        getRootRef: getRootRef,
        ...restProps
    }, slotProps === null || slotProps === void 0 ? void 0 : slotProps.root);
    const inputRest = useMergeProps({
        getRootRef: getRef,
        disabled,
        readOnly,
        required,
        autoFocus,
        id,
        name,
        value,
        accept,
        capture,
        multiple,
        form,
        onChange,
        onFocus,
        onBlur
    }, slotProps === null || slotProps === void 0 ? void 0 : slotProps.input);
    return /*#__PURE__*/ _jsxs(Button, {
        Component: "label",
        align: align,
        stretched: stretched,
        mode: mode,
        appearance: appearance,
        size: size,
        before: before,
        after: after,
        loading: loading,
        disabled: inputRest.disabled,
        onClick: withLabelClickWrapper(onClick),
        ...rootRest,
        children: [
            /*#__PURE__*/ _jsx(VisuallyHidden, {
                title: "",
                Component: "input",
                type: "file",
                ...inputRest
            }),
            children
        ]
    });
};
