// @vkontakte/vkui/dist/components/Snackbar/subcomponents/Basic/Basic.js
'use client';
import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
import { classNames } from "@vkontakte/vkjs";
import { useAdaptivity } from "../../../../hooks/useAdaptivity.js";
import { RootComponent } from "../../../RootComponent/RootComponent.js";
import { Paragraph } from "../../../Typography/Paragraph/Paragraph.js";
import { Subhead } from "../../../Typography/Subhead/Subhead.js";
const stylesLayout = {
    none: "vkuiBasic__layoutNone",
    vertical: "vkuiBasic__layoutVertical",
    horizontal: "vkuiBasic__layoutHorizontal"
};
const densityClassNames = {
    none: "vkuiBasic__densityNone",
    regular: "vkuiBasic__densityRegular"
};
export function Basic({ layout: layoutProps, action, after, before, mode, subtitle, children, ...restProps }) {
    const { density = 'none' } = useAdaptivity();
    const layout = after || subtitle ? 'vertical' : 'none';
    return /*#__PURE__*/ _jsxs(RootComponent, {
        ...restProps,
        baseClassName: classNames("vkuiBasic__body", stylesLayout[layoutProps || layout], density !== 'compact' && densityClassNames[density], mode === 'dark' && "vkuiBasic__modeDark"),
        children: [
            before && /*#__PURE__*/ _jsx("div", {
                className: "vkuiBasic__before",
                children: before
            }),
            /*#__PURE__*/ _jsxs("div", {
                className: "vkuiBasic__content",
                children: [
                    /*#__PURE__*/ _jsx(Paragraph, {
                        className: "vkuiBasic__contentText",
                        children: children
                    }),
                    subtitle && !action && /*#__PURE__*/ _jsx(Subhead, {
                        className: "vkuiBasic__contentSubtitle",
                        children: subtitle
                    }),
                    action && !subtitle && /*#__PURE__*/ _jsx("div", {
                        className: "vkuiBasic__action",
                        children: action
                    })
                ]
            }),
            after && /*#__PURE__*/ _jsx("div", {
                className: "vkuiBasic__after",
                children: after
            })
        ]
    });
}
