// @vkontakte/vkui/dist/components/CustomScrollView/CustomScrollView.js
'use client';
import { jsx as _jsx } from "react/jsx-runtime";
import { classNames } from "@vkontakte/vkjs";
const overscrollBehaviorClassNames = {
    auto: undefined,
    contain: "vkuiCustomScrollView__overscrollBehaviorContain",
    none: "vkuiCustomScrollView__overscrollBehaviorNone"
};
const scrollBehaviorClassNames = {
    auto: undefined,
    smooth: "vkuiCustomScrollView__scrollBehaviorSmooth"
};
/**
 * @see https://vkui.io/components/custom-scroll-view
 */ export const CustomScrollView = ({ className, children, enableHorizontalScroll = false, onScroll, getRootRef, overscrollBehavior = 'auto', scrollBehavior = 'auto', scrollbarHidden = false, ...restProps })=>{
    return /*#__PURE__*/ _jsx("div", {
        className: classNames(className, "vkuiCustomScrollView__host", enableHorizontalScroll && "vkuiCustomScrollView__horizontalScrollEnabled", overscrollBehaviorClassNames[overscrollBehavior], scrollBehaviorClassNames[scrollBehavior], scrollbarHidden && "vkuiCustomScrollView__scrollbarHidden"),
        ref: getRootRef,
        onScroll: onScroll,
        ...restProps,
        children: children
    });
};
