// @vkontakte/vkui/dist/components/Switch/Switch.js
'use client';
import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
import * as React from "react";
import { classNames } from "@vkontakte/vkjs";
import { useAdaptivity } from "../../hooks/useAdaptivity.js";
import { useConfigDirection } from "../../hooks/useConfigDirection.js";
import { useFocusVisible } from "../../hooks/useFocusVisible.js";
import { useFocusVisibleClassName } from "../../hooks/useFocusVisibleClassName.js";
import { useMergeProps } from "../../hooks/useMergeProps.js";
import { usePlatform } from "../../hooks/usePlatform.js";
import { callMultiple } from "../../lib/callMultiple.js";
import { warnOnce } from "../../lib/warnOnce.js";
import { withLabelClickWrapper } from "../../lib/withLabelClickWrapper.js";
import { RootComponent } from "../RootComponent/RootComponent.js";
import { VisuallyHidden } from "../VisuallyHidden/VisuallyHidden.js";
const warn = warnOnce('Switch');
const densityClassNames = {
    none: "vkuiSwitch__densityNone",
    compact: "vkuiSwitch__densityCompact"
};
/**
 * @see https://vkui.io/components/switch
 */ export const Switch = ({ getRef, // Input props
checked, defaultChecked, disabled, readOnly, required, autoFocus, id, name, value, form, onChange, onFocus, onBlur, slotProps, ...restProps })=>{
    /* istanbul ignore if: не проверяем в тестах */ if (process.env.NODE_ENV === 'development' && getRef) {
        warn('Свойство `getRef` устаревшее, используйте `slotProps={ input: { getRootRef: ... } }`');
    }
    const { onClick: onRootClick, ...rootRest } = useMergeProps(restProps, slotProps === null || slotProps === void 0 ? void 0 : slotProps.root);
    const { checked: checkedProp, onBlur: onInputBlur, onFocus: onInputFocus, onClick, className: inputClassName, ...inputRest } = useMergeProps({
        getRootRef: getRef,
        checked,
        defaultChecked,
        disabled,
        readOnly,
        required,
        autoFocus,
        id,
        name,
        value,
        form,
        onChange,
        onFocus,
        onBlur
    }, slotProps === null || slotProps === void 0 ? void 0 : slotProps.input);
    const direction = useConfigDirection();
    const isRtl = direction === 'rtl';
    const platform = usePlatform();
    const { density = 'none' } = useAdaptivity();
    const { focusVisible, onBlur: onFocusVisibleBlur, onFocus: onFocusVisibleFocus } = useFocusVisible();
    const focusVisibleClassNames = useFocusVisibleClassName({
        focusVisible,
        mode: 'outside'
    });
    const handleBlur = callMultiple(onFocusVisibleBlur, onInputBlur);
    const handleFocus = callMultiple(onFocusVisibleFocus, onInputFocus);
    const [localUncontrolledChecked, setLocalUncontrolledChecked] = React.useState(Boolean(inputRest.defaultChecked));
    const isControlled = checkedProp !== undefined;
    const syncUncontrolledCheckedStateOnClick = React.useCallback((e)=>{
        if (isControlled) {
            return;
        }
        const switchTarget = e.target;
        setLocalUncontrolledChecked(switchTarget.checked);
    }, [
        isControlled
    ]);
    const inputProps = {
        Component: 'input',
        type: 'checkbox',
        role: 'switch',
        onBlur: handleBlur,
        onFocus: handleFocus,
        onClick: callMultiple(syncUncontrolledCheckedStateOnClick, onClick),
        ...inputRest
    };
    if (isControlled) {
        inputProps.checked = checkedProp;
        inputProps['aria-checked'] = checkedProp ? 'true' : 'false';
    } else {
        inputProps['aria-checked'] = localUncontrolledChecked ? 'true' : 'false';
    }
    return /*#__PURE__*/ _jsxs(RootComponent, {
        Component: "label",
        baseClassName: classNames("vkuiSwitch__host", density !== 'regular' && densityClassNames[density], platform === 'ios' ? "vkuiSwitch__ios" : "vkuiSwitch__default", inputRest.disabled && "vkuiSwitch__disabled", isRtl && "vkuiSwitch__rtl", focusVisibleClassNames),
        onClick: withLabelClickWrapper(onRootClick),
        ...rootRest,
        children: [
            /*#__PURE__*/ _jsx(VisuallyHidden, {
                className: classNames(inputClassName, "vkuiSwitch__inputNative"),
                ...inputProps
            }),
            /*#__PURE__*/ _jsxs("span", {
                "aria-hidden": true,
                className: "vkuiSwitch__inputFake",
                children: [
                    /*#__PURE__*/ _jsx("span", {
                        className: "vkuiSwitch__track"
                    }),
                    /*#__PURE__*/ _jsx("span", {
                        "aria-hidden": true,
                        className: classNames("vkuiSwitch__handle", platform !== 'ios' && !inputRest.disabled && "vkuiSwitch__handleWithRipple")
                    })
                ]
            })
        ]
    });
};
