// @vkontakte/vkui/dist/components/CalendarTime/CalendarTimePicker.js
'use client';
import { jsx as _jsx } from "react/jsx-runtime";
import { useState } from "react";
import * as React from "react";
import { AdaptivityProvider } from "../AdaptivityProvider/AdaptivityProvider.js";
import { CustomSelect } from "../CustomSelect/CustomSelect.js";
import { CustomSelectOption } from "../CustomSelectOption/CustomSelectOption.js";
const selectFilterFn = ()=>true;
const validateValue = (value, validValues)=>{
    const numValue = Number(value);
    return !isNaN(numValue) && validValues.some((v)=>v.value === numValue);
};
/* eslint-enable jsdoc/require-jsdoc */ export const CalendarTimePicker = ({ value, getNumericValue, options, onChange, setTime, inputRef, inputLabel, inputTestId, onInputKeyDown })=>{
    const [inputValue, setInputValue] = useState(undefined);
    const onBlur = React.useCallback(()=>{
        setInputValue(undefined);
    }, []);
    const _onChange = React.useCallback((_, newValue)=>onChange === null || onChange === void 0 ? void 0 : onChange(setTime(value, Number(newValue))), [
        onChange,
        setTime,
        value
    ]);
    const onPickerValueChange = (e)=>{
        const numericValue = e.target.value.replace(/\D/g, '');
        setInputValue(numericValue);
        if (validateValue(numericValue, options)) {
            onChange === null || onChange === void 0 ? void 0 : onChange(setTime(value, Number(numericValue)));
        }
    };
    return /*#__PURE__*/ _jsx("div", {
        className: "vkuiCalendarTime__picker",
        children: /*#__PURE__*/ _jsx(AdaptivityProvider, {
            density: "compact",
            children: /*#__PURE__*/ _jsx(CustomSelect, {
                value: getNumericValue(value),
                options: options,
                onChange: _onChange,
                forceDropdownPortal: false,
                searchable: true,
                filterFn: selectFilterFn,
                onInputChange: onPickerValueChange,
                onInputKeyDown: onInputKeyDown,
                renderOption: ({ children: optionChildren, ...optionProps })=>/*#__PURE__*/ _jsx(CustomSelectOption, {
                        ...optionProps,
                        textNoWrap: true,
                        children: optionChildren
                    }),
                slotProps: {
                    input: {
                        'aria-label': inputLabel,
                        'data-testid': inputTestId,
                        'value': inputValue,
                        'getRootRef': inputRef,
                        onBlur
                    }
                }
            })
        })
    });
};
