// @vkontakte/vkui/dist/components/InputLike/InputLike.js
'use client';
import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
/* eslint-disable jsdoc/require-jsdoc */ import * as React from "react";
import { callMultiple } from "../../lib/callMultiple.js";
import { stopPropagation } from "../../lib/utils.js";
import { RootComponent } from "../RootComponent/RootComponent.js";
import { VisuallyHidden } from "../VisuallyHidden/VisuallyHidden.js";
const MASK_SYMBOL = String.fromCharCode(0x2007);
function getMaskElements(length) {
    const result = [];
    for(let index = 0; index < length; index += 1){
        result.push(/*#__PURE__*/ _jsx("span", {
            className: "vkuiInputLike__mask",
            children: MASK_SYMBOL
        }, index));
    }
    return result;
}
export const InputLike = ({ value, length, index, onElementSelect, onFocus, label, readOnly, onKeyDown, ...restProps })=>{
    var _ref;
    const handleElementSelect = React.useCallback((event)=>{
        stopPropagation(event);
        onElementSelect === null || onElementSelect === void 0 ? void 0 : onElementSelect(index);
    }, [
        index,
        onElementSelect
    ]);
    return /*#__PURE__*/ _jsxs(RootComponent, {
        Component: "span",
        baseClassName: (value === null || value === void 0 ? void 0 : value.length) === length ? "vkuiInputLike__host" : undefined,
        tabIndex: readOnly ? -1 : 0,
        onFocus: callMultiple(onFocus, handleElementSelect),
        onKeyDown: readOnly ? undefined : onKeyDown,
        ...restProps,
        children: [
            label && /*#__PURE__*/ _jsx(VisuallyHidden, {
                children: label
            }),
            value === null || value === void 0 ? void 0 : value.slice(0, length - 1),
            (value === null || value === void 0 ? void 0 : value.slice(length - 1)) && /*#__PURE__*/ _jsx("span", {
                className: "vkuiInputLike__lastCharacter",
                children: value.slice(length - 1)
            }, index),
            getMaskElements(length - ((_ref = value === null || value === void 0 ? void 0 : value.length) !== null && _ref !== void 0 ? _ref : 0))
        ]
    });
};
