// @vkontakte/vkui/dist/components/ModalOverlay/ModalOverlay.js
'use client';
import { jsx as _jsx } from "react/jsx-runtime";
import { classNames } from "@vkontakte/vkjs";
import { useExternRef } from "../../hooks/useExternRef.js";
import { useCSSTransition } from "../../lib/animation/useCSSTransition.js";
const positionClassNames = {
    absolute: "vkuiModalOverlay__hostPositionAbsolute",
    fixed: "vkuiModalOverlay__hostPositionFixed"
};
const transitionStateClassNames = {
    appear: "vkuiModalOverlay__hostStateEnter",
    appearing: "vkuiModalOverlay__hostStateEntering",
    appeared: "vkuiModalOverlay__hostStateEntered",
    enter: "vkuiModalOverlay__hostStateEnter",
    entering: "vkuiModalOverlay__hostStateEntering",
    entered: "vkuiModalOverlay__hostStateEntered",
    exit: "vkuiModalOverlay__hostStateExit",
    exiting: "vkuiModalOverlay__hostStateExiting",
    exited: "vkuiModalOverlay__hostStateExited"
};
/**
 * @private
 */ export const ModalOverlay = ({ visible = false, position = 'absolute', getRootRef, onClick, onClosed, onShowed, disableOpenAnimation, disableCloseAnimation, ...restProps })=>{
    const [transitionState, { ref, onTransitionEnd }] = useCSSTransition(visible, {
        enableAppear: !disableOpenAnimation,
        enableEnter: !disableOpenAnimation,
        enableExit: !disableCloseAnimation,
        onExited: onClosed,
        onEntered: onShowed
    });
    const handleRef = useExternRef(getRootRef, ref);
    return /*#__PURE__*/ _jsx("div", {
        ...restProps,
        ref: handleRef,
        "aria-hidden": "true",
        hidden: transitionState === 'exited',
        className: classNames("vkuiModalOverlay__host", onClick === undefined && "vkuiModalOverlay__nonInteractive", positionClassNames[position], transitionStateClassNames[transitionState]),
        onClick: onClick,
        onTransitionEnd: onTransitionEnd
    });
};
